---
title: "microbiome_tutorial"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{microbiome_tutorial}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  warning = FALSE,
  message = FALSE,
  comment = "#>"
)
```

```{r setup}
library(FAVA)
library(dplyr)
library(tidyr)
library(ggplot2)
```


In this tutorial, I visualize the relative abundances of 

I acoomplish these tasks using three files, all of which are included in the FAVA R package:

* `xue_microbiome_sample` is a dataframe containing 77 microbiome samples generated by Xue et al. (2023), each from one of three subjects (XBA, XDA, and XMA) at a series of time points between study day 1 and day 64. Each row of `xue_microbiome_sample` represents a single microbiome sample. `xue_microbiome_sample` has 526 columns: 

    * `subject` - the subject each sample corresponds to 

    * `timepoint` - the study day at which at sample was taken

    * `Actinomyces_sp_58647`, ..., `Xenorhabdus_bovienii_57960` - the relative abundance of the corresponding bacterial species (there are 524 species in total)
    
    
* `xue_species_tree` is a phylogenetic tree object describing the evolutionary relationships of nearly 6,000 bacterial species, including the 524 present in `xue_microbiome_sample`. In this tutorial, I show you how to transform such a tree into a distance matrix using the [ape package](http://ape-package.ird.fr/), and how to then transform the distance matrix into a similarity matrix that we can use an input for FAVA. 


* `xue_species_info` is a dataframe providing the taxonomic information for each species. We will use this dataframe to restrict our similarity matrix (generated from `xue_species_tree`) to the 524 species present in `xue_microbiome_sample`. 



# 1 - Visualize relative abundances

```{r}
# generate a really big color palette
set.seed(5)
pal = c(MetBrewer::met.brewer("Johnson", 542))[sample(1:524)]

# make plot
Q_plot(Q = xue_microbiome_sample, 
       K = 524, 
       group = "subject", 
       arrange = "vertical") + 
  scale_color_manual(values = rep("white", 524)) +
  scale_fill_manual(values = pal)
```

# 2 - Compute unweighted FAVA across all relative abundances for each subject

```{r}
xue_microbiome_sample %>% 
  group_by(subject) %>%
  group_map(~ fava(select(.x, - timepoint))) %>%
  `names<-`(unique(xue_microbiome_sample$subject))
```

# 3 - Compute weighted FAVA across all relative abundances for each subject

## Generate a species similarity matrix
```{r}
# First generate a distance matrix based on the full species tree
sp.dist <- ape::cophenetic.phylo(x = xue_species_tree)

# Next, restrict the species tree to the 524 species included in the data set, xue_microbiome sample
species_id_numbers = filter(xue_species_info, 
       species_id %in% colnames(xue_microbiome_sample)) %>% 
  select(species_id_number) %>% 
  unlist(use.names = FALSE) %>% 
  as.character

sp.dist.524 = sp.dist[species_id_numbers, species_id_numbers]

# Visualize S matrix
sp.dist.524 %>% data.frame %>%
  mutate(name2 = rownames(sp.dist.524)) %>% 
  pivot_longer(cols =  1:ncol(sp.dist.524)) %>% 
  ggplot() + geom_raster(aes(x = name, y = name2, fill = value)) + scale_fill_viridis_c() + theme_void()

# There are a few different ways to convert a distance matrix to a similarity matrix:
sp.sim <-  exp(-sp.dist.524)
# sp.sim <-  1/(sp.dist + 1)
# sp.sim <-  sqrt(1 - sp.dist/max(sp.dist))

```

## Use this species similarity matrix to compute FAVA for each subject
```{r}
fava_list = c()
for(sub in unique(xue_microbiome_sample$subject)){
  Q_subject = xue_microbiome_sample %>% 
    filter(subject == sub)
  
  fava_list = c(fava_list, 
fava(Q = select(Q_subject, -c(subject, timepoint)), 
     w = time_weights(Q_subject$timepoint), 
     S = sp.sim) )
}

fava_list %>% `names<-`(unique(xue_microbiome_sample$subject))
```

# 4 - Compute FAVA in sliding windows six samples wide for each subject

## Compute FAVA for each window
```{r}
window_out = window_fava(Q = xue_microbiome_sample,
                         K = 524,
                         w = time_weights(times = xue_microbiome_sample$timepoint, 
                              group = xue_microbiome_sample$subject),
                         S = sp.sim,
                         window_size = 6, 
                         group = "subject", 
                         index = "timepoint")
head(window_out)
```

## Visualize FAVA in sliding windows
```{r}
window_plot(window_out)
window_plot(window_out) + facet_wrap(~ group)
```

