<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Analysis of microbiome data with FAVA • FAVA</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Analysis of microbiome data with FAVA">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">FAVA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/microbiome_tutorial.html">Analysis of microbiome data with FAVA</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/MaikeMorrison/FAVA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<script src="microbiome_tutorial_files/kePrint-0.0.1/kePrint.js"></script><link href="microbiome_tutorial_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Analysis of microbiome data with FAVA</h1>
                        <h4 data-toc-skip class="author">Maike
Morrison</h4>
            
            <h4 data-toc-skip class="date">March 27, 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/MaikeMorrison/FAVA/blob/v1.0.9/vignettes/microbiome_tutorial.Rmd" class="external-link"><code>vignettes/microbiome_tutorial.Rmd</code></a></small>
      <div class="d-none name"><code>microbiome_tutorial.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://maikemorrison.github.io/FAVA/">FAVA</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <em>FAVA</em> R package implements the statistic FAVA, an
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>-based
Assessment of Variability across vectors of relative Abundances, as well
as a suite of helper functions which enable the visualization and
statistical analysis of relative abundance data. The <em>FAVA</em> R
package accompanies the paper, <a href="https://www.biorxiv.org/content/10.1101/2024.07.03.601929v1" class="external-link">“Quantifying
compositional variability in microbial communities with FAVA”</a> by
Maike Morrison, Katherine Xue, and Noah Rosenberg. This tutorial
provides a guide to the usage of the <em>FAVA</em> R package for the
analysis of microbiome data.</p>
<p>The <em>FAVA</em> R package includes the following core
functions:</p>
<ul>
<li><p><code>fava</code>: Compute FAVA across the rows of a relative
abundance matrix</p></li>
<li><p><code>plot_relabund</code>: Visualize a relative abundance matrix
as a stacked bar plot</p></li>
</ul>
<!-- * `bootstrap_fava`: Estimate the uncertainty in FAVA by generating bootstrap replicates of one or more relative abundance matrices and computing FAVA for each replicate matrix --><ul>
<li>
<code>window_fava</code>: Compute FAVA in sliding windows across the
rows of a relative abundance matrix</li>
</ul>
<p>The statistic FAVA summarizes the variability across the rows of a
relative abundance matrix in a single index that ranges from 0 to 1. For
typical microbiome data, the rows of each matrix represent microbiome
samples and the entries of each row represent the relative abundance of
a taxonomic category (e.g., OTU, species, or genus) or a functional
category (e.g., CAZyme or gene ortholog). Such matrices are often
referred to as “OTU tables,” and are usually inferred from 16S or
metagenomic sequencing data. You must obtain these relative abundances
from your sequencing data before using the R package <em>FAVA</em>.</p>
<p>FAVA can be used to quantify variability in many different contexts.
For example,</p>
<ul>
<li><p>If each row corresponds to a time point, FAVA represents the
temporal stability of the community.</p></li>
<li><p>If each row corresponds to a spatial sampling location, FAVA
represents the spatial heterogeneity of the community.</p></li>
<li><p>If each row corresponds to a replicate in a community assembly
experiment, FAVA represents the repeatability of the community
assembly.</p></li>
<li><p>If each row represents a distinct individual, FAVA represents the
inter-individual variability in microbiome composition.</p></li>
</ul>
<div class="section level3">
<h3 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
<p>In this tutorial, we will explain the data required to use the
<em>FAVA</em> R package (<a href="#data-specifications">Data
specifications</a>). We will then analyze example data from <a href="https://doi.org/10.1101/2023.09.26.559480" class="external-link">Xue et al. (2024)</a>
(<a href="#example-analysis">Example Analysis</a>). With this data set
and the <em>FAVA</em> R package, we visualize the microbiome composition
of three subjects over time, use FAVA to quantify the temporal
variability across these microbiome samples, and explore these dynamics
at a finer resolution using sliding windows over time.</p>
</div>
</div>
<div class="section level2">
<h2 id="data-specifications">Data specifications<a class="anchor" aria-label="anchor" href="#data-specifications"></a>
</h2>
<p>To use <em>FAVA</em>, your data must be in the form of a matrix, data
frame, or tibble with rows corresponding to samples and columns
corresponding to categories such as bacterial species. If metadata
(e.g., sample ID, time point, subject, experimental condition,
replicate, etc.) are included, these columns must be on the left-hand
side of the matrix, while the categories comprising the composition of
the sample (e.g., bacterial species) must be on the right-hand side of
the matrix. If your matrix contains metadata, you must specify
<code>K</code>, the number of categories. While these categories can
represent many things, we will for simplicity use “species” in this
document to refer to the categories comprising each relative abundance
sample.</p>
<p>Your matrix may contain samples from multiple groups you would like
to analyze separately. In this case, you must provide the name of the
column specifying the group each sample belongs to as the
<code>group</code> parameter. For example, to quantify variability
across all samples from each experiment in the matrix pictured below, we
would specify <code>group = "Experiment"</code>. You can also group by
multiple columns by providing a vector of column names. For example, if
your data had columns <code>var1</code> and <code>var2</code> you wished
to group by, you would specify <code>group = c("var1", "var2")</code>.
See the section <a href="#compute-unweighted-fava-with-multiple-groups">Compute unweighted
FAVA with multiple groups</a> for details.</p>
<p><img src="../reference/figures/schematic_data_structure_1.png" width="1000px"></p>
<p>You can read your data into R using a function such as
<code>read.csv</code>. If you use <em>phyloseq</em>, you can simply
extract and transpose (pivot) the OTU table. <em>phyloseq</em> users may
also wish to join their OTU table with their sample data table. Sample
code for these tasks is provided below. You may also want to confirm
that the right
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>K</mi><annotation encoding="application/x-tex">K</annotation></semantics></math>
rows of your matrix each sum to 1.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example code to read in a data set</span></span>
<span><span class="va">my_data</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"Path_to/my_data.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># If your relative abundances are in a phyloseq object,</span></span>
<span><span class="co"># make one object combining the sample data (left-hand side)</span></span>
<span><span class="co"># and the OTU relative abundances (right-hand side)</span></span>
<span><span class="va">my_data</span> <span class="op">=</span> <span class="fu">FAVA</span><span class="fu">::</span><span class="fu"><a href="../reference/relab_phyloseq.html">relab_phyloseq</a></span><span class="op">(</span><span class="va">phyloseq_object</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Confirm that your samples each sum to 1</span></span>
<span><span class="co"># if columns 4 through 10 contain the relevant categories</span></span>
<span><span class="co"># and columns 1, 2, and 3 contain metadata</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">my_data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Example code to convert counts to relative abundances</span></span>
<span><span class="va">my_data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span><span class="op">]</span> <span class="op">=</span> <span class="va">my_data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span><span class="op">]</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">my_data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="optional-species-similarity-matrix">Optional: species similarity matrix<a class="anchor" aria-label="anchor" href="#optional-species-similarity-matrix"></a>
</h4>
<p>In order to compute a version of the FAVA variability statistic that
accounts for the phylogenetic similarity between species, you must
provide a phylogenetic similarity matrix,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>S</mi><annotation encoding="application/x-tex">S</annotation></semantics></math>.
Such a matrix can be generated from species sequence data via the four
steps outlined below. We include example code to generate a similarity
matrix from a <em>phyloseq</em> object. This process is discussed at
greater length in a supplement to this tutorial, <a href="#generating-a-similarity-matrix">Generating a similarity
matrix</a>.</p>
<ol style="list-style-type: decimal">
<li><p><strong>Generate a phylogenetic tree describing the relationships
among the species present in your samples.</strong> There are many
platforms for such analysis, such as <a href="https://f1000research.com/articles/5-1492/v2" class="external-link">this workflow using
DADA2</a>.</p></li>
<li><p><strong>Convert the phylogenetic tree to a distance
matrix.</strong> Phylogenetic tree objects can be easily converted to
distance matrices using functions such as <code>cophenetic.phylo</code>
in the <em>ape</em> R package. If
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>D</mi><annotation encoding="application/x-tex">D</annotation></semantics></math>
is the phylogenetic distance matrix, entry
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><annotation encoding="application/x-tex">D_{i,j}</annotation></semantics></math>
represents the phylogenetic distance between species
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>
and species
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>j</mi><annotation encoding="application/x-tex">j</annotation></semantics></math>.</p></li>
<li><p><strong>Convert the phylogenetic distance matrix to a similarity
matrix.</strong> Whereas the distance between two identical species is
0, the similarity between two identical species is 1. We therefore use a
simple transformation to convert the distance between species
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>j</mi><annotation encoding="application/x-tex">j</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><annotation encoding="application/x-tex">D_{i,j}</annotation></semantics></math>,
to the similarity between species
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>j</mi><annotation encoding="application/x-tex">j</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>S</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><annotation encoding="application/x-tex">S_{i,j}</annotation></semantics></math>,
mapping distances of 0 to similarities of 1. For the example analysis in
this tutorial and in the accompanying paper, we use the transformation
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>S</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>=</mo><mo>exp</mo><mrow><mo stretchy="true" form="prefix">(</mo><mo>−</mo><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">S_{i,j}=\exp(-D_{i,j})</annotation></semantics></math>.
Other transformations, such as
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>S</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>=</mo><mfrac><mn>1</mn><mrow><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>+</mo><mn>1</mn></mrow></mfrac></mrow><annotation encoding="application/x-tex">S_{i,j}=\frac{1}{D_{i,j}+1}</annotation></semantics></math>
or
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>S</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>=</mo><mn>1</mn><mo>−</mo><mfrac><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mrow><mo>max</mo><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac></mrow><annotation encoding="application/x-tex">S_{i,j}=1-\frac{D_{i,j}}{\max{(D_{i,j})}}</annotation></semantics></math>,
are also suitable but result in a different mean similarity across
species. The choice of transformation is discussed at length in a
supplement to this tutorial, <a href="#generating-a-similarity-matrix">Generating a similarity
matrix</a>.</p></li>
<li><p><strong>Ensure that the order and identity of the species in the
similarity matrix match the order and identity of the species in your
relative abundance matrix.</strong></p></li>
</ol>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># (1)</span></span>
<span><span class="co"># Here, we assume that you have already generated a phylogenetic tree</span></span>
<span><span class="co"># and that it is a part of your phyloseq object.</span></span>
<span><span class="va">tree</span> <span class="op">=</span> <span class="fu">phy_tree</span><span class="op">(</span><span class="va">phyloseq_object</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># (2)</span></span>
<span><span class="va">distance_matrix</span> <span class="op">=</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/cophenetic.phylo.html" class="external-link">cophenetic.phylo</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># (3)</span></span>
<span><span class="co"># alternative similarity matrices:</span></span>
<span><span class="va">similarity_matrix</span> <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="op">(</span><span class="va">distance_matrix</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">similarity_matrix</span> <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">distance_matrix</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">distance_matrix</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># the similarity matrix we use:</span></span>
<span><span class="va">similarity_matrix</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span><span class="op">-</span><span class="va">distance_matrix</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># (4)</span></span>
<span><span class="co"># Get the names of the species in your relative abundance matrix</span></span>
<span><span class="va">species_order</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">my_data</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Confirm that the entries of the similarity matrix</span></span>
<span><span class="co"># correspond to relative abundance matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="va">species_order</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">similarity_matrix</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="va">species_order</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">similarity_matrix</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># If they do not, you can re-order the rows and columns of</span></span>
<span><span class="co"># your similarity matrix to match your data:</span></span>
<span><span class="va">similarity_matrix_reordered</span> <span class="op">=</span> <span class="va">similarity_matrix</span><span class="op">[</span><span class="va">species_order</span>, <span class="va">species_order</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># confirm that all diagonal elements are still 1</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">similarity_matrix_reordered</span><span class="op">)</span></span></code></pre></div>
<p>Having generated a properly formatted relative abundance matrix, and
possibly a species similarity matrix as well, it’s time to use
<em>FAVA</em>!</p>
</div>
</div>
<div class="section level2">
<h2 id="example-analysis">Example analysis<a class="anchor" aria-label="anchor" href="#example-analysis"></a>
</h2>
<p>As a guide for the application of <em>FAVA</em> to microbiome data,
we demonstrate each of the package’s core functions using example data
generated by <a href="https://doi.org/10.1101/2023.09.26.559480" class="external-link">Xue et
al. (2024)</a>. This data set contains time series microbiome samples
from three human subjects who each took an antibiotic midway through the
study period (<a href="#example-data">0 - Example data</a>). We first
visualize the composition of these subject’s microbiome communities over
time (<a href="#visualize-relative-abundances">1 - Visualize relative
abundances</a>). For each subject, we next compute the total variability
across the study period (<a href="#compute-unweighted-fava">2 - Compute
unweighted FAVA</a> and <a href="#compute-weighted-fava">3 - Compute
weighted FAVA</a>). We also estimate the uncertainty in these
variability measures, which allows us to perform statistical comparisons
of each subject’s temporal microbiome variability (<a href="#bootstrapping">4 - Bootstrapping</a>). Finally, we explore how
compositional variability changes over time using a sliding window
analysis (<a href="#sliding-windows">5 - Sliding windows</a>).</p>
<div class="section level3">
<h3 id="example-data">0 - Example data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h3>
<p>In this tutorial, we analyze longitudinal microbiome composition data
generated by <a href="https://doi.org/10.1101/2023.09.26.559480" class="external-link">Xue et
al. (2024)</a>, data that is also analyzed in the paper by Morrison et
al. For example analyses, a subset of this data is provided under the
name <code>xue_microbiome_sample</code> in the <em>FAVA</em> R package.
<code>xue_microbiome_sample</code> contains the relative abundances of
bacterial species in samples from three subjects: XBA, XDA, and XMA.
Each subject collected weekly samples for four weeks before and after a
three week window of daily sampling, the middle of which contained a
one-week antibiotic course (depicted below; dots correspond to sampling
days, yellow dots correspond to sampling days coinciding with the
antibiotic).
<img src="microbiome_tutorial_files/figure-html/unnamed-chunk-5-1.png" width="768"></p>
<p>Each row of <code>xue_microbiome_sample</code> represents a single
microbiome sample. <code>xue_microbiome_sample</code> has 526
columns:</p>
<ul>
<li><p><code>subject</code>: the subject each sample corresponds
to</p></li>
<li><p><code>timepoint</code>: the study day when each sample was
collected</p></li>
<li><p><code>Actinomyces_sp_58647</code>, …,
<code>Xenorhabdus_bovienii_57960</code>: the relative abundance of the
corresponding bacterial species (there are 524 species in
total)</p></li>
</ul>
<p>You can explore the structure of this data using the following
functions:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># open the data set in a new window</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/View.html" class="external-link">View</a></span><span class="op">(</span><span class="va">xue_microbiome_sample</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># view the structure of the data set</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">xue_microbiome_sample</span><span class="op">)</span></span></code></pre></div>
Here are the first 40 rows and the first 20 columns of the relative
abundance matrix, <code>xue_microbiome_sample</code>:
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:800px; ">
<table class="table">
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
subject
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
timepoint
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Actinomyces_sp_58647
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Actinomyces_sp_59161
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Actinomyces_sp_59944
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Aggregatibacter_aphrophilus_58143
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Alistipes_finegoldii_56071
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Anaerostipes_hadrus_55206
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_acidifaciens_59693
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_caccae_53434
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_cellulosilyticus_58046
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_clarus_62282
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_faecis_58503
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_finegoldii_57739
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_fluxus_62283
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_fragilis_54507
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_graminisolvens_48736
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_intestinalis_61596
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_massiliensis_44749
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_nordii_55557
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
3.00e-05
</td>
<td style="text-align:right;">
1.95e-05
</td>
<td style="text-align:right;">
0.0000402
</td>
<td style="text-align:right;">
0.0000932
</td>
<td style="text-align:right;">
0.0105169
</td>
<td style="text-align:right;">
0.0047967
</td>
<td style="text-align:right;">
0.0005821
</td>
<td style="text-align:right;">
0.0000417
</td>
<td style="text-align:right;">
0.0490133
</td>
<td style="text-align:right;">
1.04e-05
</td>
<td style="text-align:right;">
0.1053265
</td>
<td style="text-align:right;">
0.0345826
</td>
<td style="text-align:right;">
7.2e-06
</td>
<td style="text-align:right;">
0.0030965
</td>
<td style="text-align:right;">
2.65e-05
</td>
<td style="text-align:right;">
0.0017469
</td>
<td style="text-align:right;">
0.0000383
</td>
<td style="text-align:right;">
9.8e-06
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0238779
</td>
<td style="text-align:right;">
0.0062339
</td>
<td style="text-align:right;">
0.0008724
</td>
<td style="text-align:right;">
0.0000121
</td>
<td style="text-align:right;">
0.0563169
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0642161
</td>
<td style="text-align:right;">
0.0303697
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0050509
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0023815
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0792188
</td>
<td style="text-align:right;">
0.0135462
</td>
<td style="text-align:right;">
0.0013047
</td>
<td style="text-align:right;">
0.0000689
</td>
<td style="text-align:right;">
0.1457363
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0491287
</td>
<td style="text-align:right;">
0.0133849
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0052595
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0059996
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0371471
</td>
<td style="text-align:right;">
0.0050842
</td>
<td style="text-align:right;">
0.0016665
</td>
<td style="text-align:right;">
0.0001235
</td>
<td style="text-align:right;">
0.2189861
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.1006940
</td>
<td style="text-align:right;">
0.0119472
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0019313
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0078913
</td>
<td style="text-align:right;">
0.0001656
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0421698
</td>
<td style="text-align:right;">
0.0035287
</td>
<td style="text-align:right;">
0.0011471
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.1506156
</td>
<td style="text-align:right;">
6.55e-05
</td>
<td style="text-align:right;">
0.0867622
</td>
<td style="text-align:right;">
0.0079372
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0021749
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0066130
</td>
<td style="text-align:right;">
0.0002462
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
4.30e-05
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0455694
</td>
<td style="text-align:right;">
0.0011405
</td>
<td style="text-align:right;">
0.0011890
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.1668561
</td>
<td style="text-align:right;">
1.66e-05
</td>
<td style="text-align:right;">
0.0960336
</td>
<td style="text-align:right;">
0.0157613
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0066761
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0071822
</td>
<td style="text-align:right;">
0.0001485
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000513
</td>
<td style="text-align:right;">
0.0215852
</td>
<td style="text-align:right;">
0.0029459
</td>
<td style="text-align:right;">
0.0004581
</td>
<td style="text-align:right;">
0.0000524
</td>
<td style="text-align:right;">
0.1380022
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0737215
</td>
<td style="text-align:right;">
0.0035850
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0039929
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0054444
</td>
<td style="text-align:right;">
0.0000470
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000555
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0265527
</td>
<td style="text-align:right;">
0.0040331
</td>
<td style="text-align:right;">
0.0002825
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.1328648
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0764246
</td>
<td style="text-align:right;">
0.0036098
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0050594
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0054381
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0336038
</td>
<td style="text-align:right;">
0.0031125
</td>
<td style="text-align:right;">
0.0009802
</td>
<td style="text-align:right;">
0.0002095
</td>
<td style="text-align:right;">
0.1993128
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0898991
</td>
<td style="text-align:right;">
0.0098039
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0053260
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0078840
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
28
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0519917
</td>
<td style="text-align:right;">
0.0034521
</td>
<td style="text-align:right;">
0.0008764
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.1730094
</td>
<td style="text-align:right;">
1.82e-05
</td>
<td style="text-align:right;">
0.0713686
</td>
<td style="text-align:right;">
0.0061653
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0044282
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0081456
</td>
<td style="text-align:right;">
0.0001015
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
8.90e-06
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000083
</td>
<td style="text-align:right;">
0.0595329
</td>
<td style="text-align:right;">
0.0025951
</td>
<td style="text-align:right;">
0.0005863
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.1697048
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0816831
</td>
<td style="text-align:right;">
0.0048515
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0045954
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0083918
</td>
<td style="text-align:right;">
0.0000705
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0162976
</td>
<td style="text-align:right;">
0.0094940
</td>
<td style="text-align:right;">
0.0008652
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.1908234
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.1224815
</td>
<td style="text-align:right;">
0.0101387
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0035144
</td>
<td style="text-align:right;">
2.90e-05
</td>
<td style="text-align:right;">
0.0068542
</td>
<td style="text-align:right;">
0.0000495
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
3.16e-05
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0328107
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
37
</td>
<td style="text-align:right;">
2.52e-05
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000426
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0015335
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.5101771
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
39
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000435
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.7516983
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
40
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0001728
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.7931325
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
41
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.6962731
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
42
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0001268
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.6934502
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
44
</td>
<td style="text-align:right;">
7.39e-05
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000366
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0896030
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0033525
</td>
<td style="text-align:right;">
0.1750438
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.5043138
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0044266
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
45
</td>
<td style="text-align:right;">
1.61e-05
</td>
<td style="text-align:right;">
1.63e-05
</td>
<td style="text-align:right;">
0.0000162
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000034
</td>
<td style="text-align:right;">
0.0000116
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.1453868
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0181264
</td>
<td style="text-align:right;">
0.1783301
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.3944012
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0053313
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
46
</td>
<td style="text-align:right;">
1.12e-05
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000112
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000122
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.2137486
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0074789
</td>
<td style="text-align:right;">
0.2534493
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.3564888
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0085713
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
47
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0003333
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.6358201
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
48
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000268
</td>
<td style="text-align:right;">
0.0005974
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000456
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0007131
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.4082404
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
9.10e-06
</td>
<td style="text-align:right;">
9.10e-06
</td>
<td style="text-align:right;">
0.0001315
</td>
<td style="text-align:right;">
0.0001872
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000288
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000098
</td>
<td style="text-align:right;">
0.0014786
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.3987532
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000098
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
50
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
8.90e-05
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0009580
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000234
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0002207
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0010833
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.5225119
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
53
</td>
<td style="text-align:right;">
1.50e-05
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000208
</td>
<td style="text-align:right;">
0.0010277
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000345
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0021639
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0093100
</td>
<td style="text-align:right;">
0.0506245
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.7725054
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000322
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
58
</td>
<td style="text-align:right;">
9.32e-05
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000909
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000221
</td>
<td style="text-align:right;">
0.0002317
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0103252
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0061615
</td>
<td style="text-align:right;">
0.0192659
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.4728604
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0001006
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
64
</td>
<td style="text-align:right;">
1.44e-05
</td>
<td style="text-align:right;">
5.09e-05
</td>
<td style="text-align:right;">
0.0000438
</td>
<td style="text-align:right;">
0.0002680
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000637
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000204
</td>
<td style="text-align:right;">
0.0003484
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.4207172
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000077
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XDA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000049
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0011234
</td>
<td style="text-align:right;">
0.0012049
</td>
<td style="text-align:right;">
0.0003451
</td>
<td style="text-align:right;">
0.0044522
</td>
<td style="text-align:right;">
0.0004768
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000315
</td>
<td style="text-align:right;">
0.0000589
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0037165
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000024
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XDA
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0021824
</td>
<td style="text-align:right;">
0.0006931
</td>
<td style="text-align:right;">
0.0000936
</td>
<td style="text-align:right;">
0.0020628
</td>
<td style="text-align:right;">
0.0002113
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000467
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0082589
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XDA
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0058961
</td>
<td style="text-align:right;">
0.0014777
</td>
<td style="text-align:right;">
0.0000147
</td>
<td style="text-align:right;">
0.0048266
</td>
<td style="text-align:right;">
0.0002718
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0001212
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0072035
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XDA
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0092503
</td>
<td style="text-align:right;">
0.0020800
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0074155
</td>
<td style="text-align:right;">
0.0012629
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000754
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0090826
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XDA
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0064540
</td>
<td style="text-align:right;">
0.0022805
</td>
<td style="text-align:right;">
0.0007913
</td>
<td style="text-align:right;">
0.0048810
</td>
<td style="text-align:right;">
0.0009828
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0001911
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0092500
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XDA
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0077478
</td>
<td style="text-align:right;">
0.0014936
</td>
<td style="text-align:right;">
0.0006939
</td>
<td style="text-align:right;">
0.0065825
</td>
<td style="text-align:right;">
0.0026588
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0001063
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0103214
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000563
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XDA
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0069048
</td>
<td style="text-align:right;">
0.0012383
</td>
<td style="text-align:right;">
0.0001930
</td>
<td style="text-align:right;">
0.0033168
</td>
<td style="text-align:right;">
0.0026463
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0001872
</td>
<td style="text-align:right;">
0.0000667
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0086995
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000709
</td>
<td style="text-align:right;">
0.0000098
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XDA
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0067567
</td>
<td style="text-align:right;">
0.0029648
</td>
<td style="text-align:right;">
0.0002057
</td>
<td style="text-align:right;">
0.0032117
</td>
<td style="text-align:right;">
0.0024584
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0099654
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0001114
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XDA
</td>
<td style="text-align:right;">
28
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0023716
</td>
<td style="text-align:right;">
0.0014425
</td>
<td style="text-align:right;">
0.0007604
</td>
<td style="text-align:right;">
0.0064171
</td>
<td style="text-align:right;">
0.0018654
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000708
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0231366
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000717
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XDA
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0027106
</td>
<td style="text-align:right;">
0.0006663
</td>
<td style="text-align:right;">
0.0002760
</td>
<td style="text-align:right;">
0.0062014
</td>
<td style="text-align:right;">
0.0012979
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000664
</td>
<td style="text-align:right;">
0.0000265
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0140499
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000620
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XDA
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0064297
</td>
<td style="text-align:right;">
0.0010045
</td>
<td style="text-align:right;">
0.0001474
</td>
<td style="text-align:right;">
0.0076271
</td>
<td style="text-align:right;">
0.0030427
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000654
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0213590
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000714
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XDA
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0265718
</td>
<td style="text-align:right;">
0.0070122
</td>
<td style="text-align:right;">
0.0003038
</td>
<td style="text-align:right;">
0.0009442
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0013056
</td>
<td style="text-align:right;">
0.0002010
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0120045
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
</tr>
</tbody>
</table>
</div>
We also provide in the <em>FAVA</em> R package a pairwise similarity
matrix, <code>xue_species_similarity</code>, which contains the
phylogenetic similarity of every pair of species included in
<code>xue_microbiome_sample</code>. Here are the first 20 rows and the
first 20 columns of our example pairwise similarity matrix,
<code>xue_species_similarity</code>:
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:800px; ">
<table class="table">
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Actinomyces_sp_58647
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Actinomyces_sp_59161
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Actinomyces_sp_59944
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Aggregatibacter_aphrophilus_58143
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Alistipes_finegoldii_56071
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Anaerostipes_hadrus_55206
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_acidifaciens_59693
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_caccae_53434
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_cellulosilyticus_58046
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_clarus_62282
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_faecis_58503
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_finegoldii_57739
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_fluxus_62283
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_fragilis_54507
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_graminisolvens_48736
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_intestinalis_61596
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_massiliensis_44749
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_nordii_55557
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_ovatus_58035
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_rodentium_59708
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Actinomyces_sp_58647
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.9924685
</td>
<td style="text-align:right;">
0.9154130
</td>
<td style="text-align:right;">
0.0663547
</td>
<td style="text-align:right;">
0.0560641
</td>
<td style="text-align:right;">
0.0770917
</td>
<td style="text-align:right;">
0.0589857
</td>
<td style="text-align:right;">
0.0609403
</td>
<td style="text-align:right;">
0.0563671
</td>
<td style="text-align:right;">
0.0583538
</td>
<td style="text-align:right;">
0.0607973
</td>
<td style="text-align:right;">
0.0605225
</td>
<td style="text-align:right;">
0.0583854
</td>
<td style="text-align:right;">
0.0626971
</td>
<td style="text-align:right;">
0.0611650
</td>
<td style="text-align:right;">
0.0592595
</td>
<td style="text-align:right;">
0.0641456
</td>
<td style="text-align:right;">
0.0621894
</td>
<td style="text-align:right;">
0.0604276
</td>
<td style="text-align:right;">
0.0582396
</td>
</tr>
<tr>
<td style="text-align:left;">
Actinomyces_sp_59161
</td>
<td style="text-align:right;">
0.9924685
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.9144248
</td>
<td style="text-align:right;">
0.0662831
</td>
<td style="text-align:right;">
0.0560036
</td>
<td style="text-align:right;">
0.0770085
</td>
<td style="text-align:right;">
0.0589220
</td>
<td style="text-align:right;">
0.0608746
</td>
<td style="text-align:right;">
0.0563062
</td>
<td style="text-align:right;">
0.0582909
</td>
<td style="text-align:right;">
0.0607317
</td>
<td style="text-align:right;">
0.0604572
</td>
<td style="text-align:right;">
0.0583223
</td>
<td style="text-align:right;">
0.0626294
</td>
<td style="text-align:right;">
0.0610990
</td>
<td style="text-align:right;">
0.0591955
</td>
<td style="text-align:right;">
0.0640763
</td>
<td style="text-align:right;">
0.0621223
</td>
<td style="text-align:right;">
0.0603623
</td>
<td style="text-align:right;">
0.0581767
</td>
</tr>
<tr>
<td style="text-align:left;">
Actinomyces_sp_59944
</td>
<td style="text-align:right;">
0.9154130
</td>
<td style="text-align:right;">
0.9144248
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0665448
</td>
<td style="text-align:right;">
0.0562247
</td>
<td style="text-align:right;">
0.0773125
</td>
<td style="text-align:right;">
0.0591547
</td>
<td style="text-align:right;">
0.0611149
</td>
<td style="text-align:right;">
0.0565285
</td>
<td style="text-align:right;">
0.0585210
</td>
<td style="text-align:right;">
0.0609714
</td>
<td style="text-align:right;">
0.0606958
</td>
<td style="text-align:right;">
0.0585526
</td>
<td style="text-align:right;">
0.0628767
</td>
<td style="text-align:right;">
0.0613402
</td>
<td style="text-align:right;">
0.0594292
</td>
<td style="text-align:right;">
0.0643293
</td>
<td style="text-align:right;">
0.0623676
</td>
<td style="text-align:right;">
0.0606006
</td>
<td style="text-align:right;">
0.0584064
</td>
</tr>
<tr>
<td style="text-align:left;">
Aggregatibacter_aphrophilus_58143
</td>
<td style="text-align:right;">
0.0663547
</td>
<td style="text-align:right;">
0.0662831
</td>
<td style="text-align:right;">
0.0665448
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0668035
</td>
<td style="text-align:right;">
0.0808160
</td>
<td style="text-align:right;">
0.0702848
</td>
<td style="text-align:right;">
0.0726138
</td>
<td style="text-align:right;">
0.0671645
</td>
<td style="text-align:right;">
0.0695319
</td>
<td style="text-align:right;">
0.0724434
</td>
<td style="text-align:right;">
0.0721160
</td>
<td style="text-align:right;">
0.0695694
</td>
<td style="text-align:right;">
0.0747071
</td>
<td style="text-align:right;">
0.0728815
</td>
<td style="text-align:right;">
0.0706110
</td>
<td style="text-align:right;">
0.0764331
</td>
<td style="text-align:right;">
0.0741022
</td>
<td style="text-align:right;">
0.0720028
</td>
<td style="text-align:right;">
0.0693957
</td>
</tr>
<tr>
<td style="text-align:left;">
Alistipes_finegoldii_56071
</td>
<td style="text-align:right;">
0.0560641
</td>
<td style="text-align:right;">
0.0560036
</td>
<td style="text-align:right;">
0.0562247
</td>
<td style="text-align:right;">
0.0668035
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0682826
</td>
<td style="text-align:right;">
0.3264756
</td>
<td style="text-align:right;">
0.3372941
</td>
<td style="text-align:right;">
0.3119819
</td>
<td style="text-align:right;">
0.3229783
</td>
<td style="text-align:right;">
0.3365024
</td>
<td style="text-align:right;">
0.3349815
</td>
<td style="text-align:right;">
0.3231528
</td>
<td style="text-align:right;">
0.3470175
</td>
<td style="text-align:right;">
0.3385376
</td>
<td style="text-align:right;">
0.3279907
</td>
<td style="text-align:right;">
0.3550346
</td>
<td style="text-align:right;">
0.3442077
</td>
<td style="text-align:right;">
0.3344560
</td>
<td style="text-align:right;">
0.3223459
</td>
</tr>
<tr>
<td style="text-align:left;">
Anaerostipes_hadrus_55206
</td>
<td style="text-align:right;">
0.0770917
</td>
<td style="text-align:right;">
0.0770085
</td>
<td style="text-align:right;">
0.0773125
</td>
<td style="text-align:right;">
0.0808160
</td>
<td style="text-align:right;">
0.0682826
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0718410
</td>
<td style="text-align:right;">
0.0742216
</td>
<td style="text-align:right;">
0.0686517
</td>
<td style="text-align:right;">
0.0710714
</td>
<td style="text-align:right;">
0.0740474
</td>
<td style="text-align:right;">
0.0737127
</td>
<td style="text-align:right;">
0.0711098
</td>
<td style="text-align:right;">
0.0763612
</td>
<td style="text-align:right;">
0.0744952
</td>
<td style="text-align:right;">
0.0721744
</td>
<td style="text-align:right;">
0.0781254
</td>
<td style="text-align:right;">
0.0757429
</td>
<td style="text-align:right;">
0.0735971
</td>
<td style="text-align:right;">
0.0709323
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_acidifaciens_59693
</td>
<td style="text-align:right;">
0.0589857
</td>
<td style="text-align:right;">
0.0589220
</td>
<td style="text-align:right;">
0.0591547
</td>
<td style="text-align:right;">
0.0702848
</td>
<td style="text-align:right;">
0.3264756
</td>
<td style="text-align:right;">
0.0718410
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.9391876
</td>
<td style="text-align:right;">
0.8081481
</td>
<td style="text-align:right;">
0.8366328
</td>
<td style="text-align:right;">
0.9254363
</td>
<td style="text-align:right;">
0.9420094
</td>
<td style="text-align:right;">
0.8370847
</td>
<td style="text-align:right;">
0.8695495
</td>
<td style="text-align:right;">
0.7947402
</td>
<td style="text-align:right;">
0.8496167
</td>
<td style="text-align:right;">
0.7336377
</td>
<td style="text-align:right;">
0.8723801
</td>
<td style="text-align:right;">
0.9476879
</td>
<td style="text-align:right;">
0.8349946
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_caccae_53434
</td>
<td style="text-align:right;">
0.0609403
</td>
<td style="text-align:right;">
0.0608746
</td>
<td style="text-align:right;">
0.0611149
</td>
<td style="text-align:right;">
0.0726138
</td>
<td style="text-align:right;">
0.3372941
</td>
<td style="text-align:right;">
0.0742216
</td>
<td style="text-align:right;">
0.9391876
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.8349278
</td>
<td style="text-align:right;">
0.8643565
</td>
<td style="text-align:right;">
0.9561026
</td>
<td style="text-align:right;">
0.9636569
</td>
<td style="text-align:right;">
0.8648234
</td>
<td style="text-align:right;">
0.8983640
</td>
<td style="text-align:right;">
0.8210757
</td>
<td style="text-align:right;">
0.8777706
</td>
<td style="text-align:right;">
0.7579484
</td>
<td style="text-align:right;">
0.9012884
</td>
<td style="text-align:right;">
0.9621451
</td>
<td style="text-align:right;">
0.8626640
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_cellulosilyticus_58046
</td>
<td style="text-align:right;">
0.0563671
</td>
<td style="text-align:right;">
0.0563062
</td>
<td style="text-align:right;">
0.0565285
</td>
<td style="text-align:right;">
0.0671645
</td>
<td style="text-align:right;">
0.3119819
</td>
<td style="text-align:right;">
0.0686517
</td>
<td style="text-align:right;">
0.8081481
</td>
<td style="text-align:right;">
0.8349278
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.8662861
</td>
<td style="text-align:right;">
0.8329680
</td>
<td style="text-align:right;">
0.8292032
</td>
<td style="text-align:right;">
0.8667541
</td>
<td style="text-align:right;">
0.8309464
</td>
<td style="text-align:right;">
0.7594582
</td>
<td style="text-align:right;">
0.9351578
</td>
<td style="text-align:right;">
0.7010683
</td>
<td style="text-align:right;">
0.8336514
</td>
<td style="text-align:right;">
0.8279024
</td>
<td style="text-align:right;">
0.8645899
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_clarus_62282
</td>
<td style="text-align:right;">
0.0583538
</td>
<td style="text-align:right;">
0.0582909
</td>
<td style="text-align:right;">
0.0585210
</td>
<td style="text-align:right;">
0.0695319
</td>
<td style="text-align:right;">
0.3229783
</td>
<td style="text-align:right;">
0.0710714
</td>
<td style="text-align:right;">
0.8366328
</td>
<td style="text-align:right;">
0.8643565
</td>
<td style="text-align:right;">
0.8662861
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.8623276
</td>
<td style="text-align:right;">
0.8584301
</td>
<td style="text-align:right;">
0.9334387
</td>
<td style="text-align:right;">
0.8602347
</td>
<td style="text-align:right;">
0.7862268
</td>
<td style="text-align:right;">
0.9107380
</td>
<td style="text-align:right;">
0.7257788
</td>
<td style="text-align:right;">
0.8630350
</td>
<td style="text-align:right;">
0.8570834
</td>
<td style="text-align:right;">
0.9311080
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_faecis_58503
</td>
<td style="text-align:right;">
0.0607973
</td>
<td style="text-align:right;">
0.0607317
</td>
<td style="text-align:right;">
0.0609714
</td>
<td style="text-align:right;">
0.0724434
</td>
<td style="text-align:right;">
0.3365024
</td>
<td style="text-align:right;">
0.0740474
</td>
<td style="text-align:right;">
0.9254363
</td>
<td style="text-align:right;">
0.9561026
</td>
<td style="text-align:right;">
0.8329680
</td>
<td style="text-align:right;">
0.8623276
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.9495472
</td>
<td style="text-align:right;">
0.8627934
</td>
<td style="text-align:right;">
0.8962553
</td>
<td style="text-align:right;">
0.8191484
</td>
<td style="text-align:right;">
0.8757103
</td>
<td style="text-align:right;">
0.7561693
</td>
<td style="text-align:right;">
0.8991728
</td>
<td style="text-align:right;">
0.9480576
</td>
<td style="text-align:right;">
0.8606391
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_finegoldii_57739
</td>
<td style="text-align:right;">
0.0605225
</td>
<td style="text-align:right;">
0.0604572
</td>
<td style="text-align:right;">
0.0606958
</td>
<td style="text-align:right;">
0.0721160
</td>
<td style="text-align:right;">
0.3349815
</td>
<td style="text-align:right;">
0.0737127
</td>
<td style="text-align:right;">
0.9420094
</td>
<td style="text-align:right;">
0.9636569
</td>
<td style="text-align:right;">
0.8292032
</td>
<td style="text-align:right;">
0.8584301
</td>
<td style="text-align:right;">
0.9495472
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.8588938
</td>
<td style="text-align:right;">
0.8922044
</td>
<td style="text-align:right;">
0.8154461
</td>
<td style="text-align:right;">
0.8717523
</td>
<td style="text-align:right;">
0.7527516
</td>
<td style="text-align:right;">
0.8951088
</td>
<td style="text-align:right;">
0.9650359
</td>
<td style="text-align:right;">
0.8567492
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_fluxus_62283
</td>
<td style="text-align:right;">
0.0583854
</td>
<td style="text-align:right;">
0.0583223
</td>
<td style="text-align:right;">
0.0585526
</td>
<td style="text-align:right;">
0.0695694
</td>
<td style="text-align:right;">
0.3231528
</td>
<td style="text-align:right;">
0.0711098
</td>
<td style="text-align:right;">
0.8370847
</td>
<td style="text-align:right;">
0.8648234
</td>
<td style="text-align:right;">
0.8667541
</td>
<td style="text-align:right;">
0.9334387
</td>
<td style="text-align:right;">
0.8627934
</td>
<td style="text-align:right;">
0.8588938
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.8606994
</td>
<td style="text-align:right;">
0.7866515
</td>
<td style="text-align:right;">
0.9112299
</td>
<td style="text-align:right;">
0.7261708
</td>
<td style="text-align:right;">
0.8635012
</td>
<td style="text-align:right;">
0.8575464
</td>
<td style="text-align:right;">
0.9399769
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_fragilis_54507
</td>
<td style="text-align:right;">
0.0626971
</td>
<td style="text-align:right;">
0.0626294
</td>
<td style="text-align:right;">
0.0628767
</td>
<td style="text-align:right;">
0.0747071
</td>
<td style="text-align:right;">
0.3470175
</td>
<td style="text-align:right;">
0.0763612
</td>
<td style="text-align:right;">
0.8695495
</td>
<td style="text-align:right;">
0.8983640
</td>
<td style="text-align:right;">
0.8309464
</td>
<td style="text-align:right;">
0.8602347
</td>
<td style="text-align:right;">
0.8962553
</td>
<td style="text-align:right;">
0.8922044
</td>
<td style="text-align:right;">
0.8606994
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.8447454
</td>
<td style="text-align:right;">
0.8735849
</td>
<td style="text-align:right;">
0.7797983
</td>
<td style="text-align:right;">
0.9167779
</td>
<td style="text-align:right;">
0.8908048
</td>
<td style="text-align:right;">
0.8585503
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_graminisolvens_48736
</td>
<td style="text-align:right;">
0.0611650
</td>
<td style="text-align:right;">
0.0610990
</td>
<td style="text-align:right;">
0.0613402
</td>
<td style="text-align:right;">
0.0728815
</td>
<td style="text-align:right;">
0.3385376
</td>
<td style="text-align:right;">
0.0744952
</td>
<td style="text-align:right;">
0.7947402
</td>
<td style="text-align:right;">
0.8210757
</td>
<td style="text-align:right;">
0.7594582
</td>
<td style="text-align:right;">
0.7862268
</td>
<td style="text-align:right;">
0.8191484
</td>
<td style="text-align:right;">
0.8154461
</td>
<td style="text-align:right;">
0.7866515
</td>
<td style="text-align:right;">
0.8447454
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.7984284
</td>
<td style="text-align:right;">
0.7607428
</td>
<td style="text-align:right;">
0.8379055
</td>
<td style="text-align:right;">
0.8141668
</td>
<td style="text-align:right;">
0.7846873
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_intestinalis_61596
</td>
<td style="text-align:right;">
0.0592595
</td>
<td style="text-align:right;">
0.0591955
</td>
<td style="text-align:right;">
0.0594292
</td>
<td style="text-align:right;">
0.0706110
</td>
<td style="text-align:right;">
0.3279907
</td>
<td style="text-align:right;">
0.0721744
</td>
<td style="text-align:right;">
0.8496167
</td>
<td style="text-align:right;">
0.8777706
</td>
<td style="text-align:right;">
0.9351578
</td>
<td style="text-align:right;">
0.9107380
</td>
<td style="text-align:right;">
0.8757103
</td>
<td style="text-align:right;">
0.8717523
</td>
<td style="text-align:right;">
0.9112299
</td>
<td style="text-align:right;">
0.8735849
</td>
<td style="text-align:right;">
0.7984284
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.7370423
</td>
<td style="text-align:right;">
0.8764286
</td>
<td style="text-align:right;">
0.8703847
</td>
<td style="text-align:right;">
0.9089547
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_massiliensis_44749
</td>
<td style="text-align:right;">
0.0641456
</td>
<td style="text-align:right;">
0.0640763
</td>
<td style="text-align:right;">
0.0643293
</td>
<td style="text-align:right;">
0.0764331
</td>
<td style="text-align:right;">
0.3550346
</td>
<td style="text-align:right;">
0.0781254
</td>
<td style="text-align:right;">
0.7336377
</td>
<td style="text-align:right;">
0.7579484
</td>
<td style="text-align:right;">
0.7010683
</td>
<td style="text-align:right;">
0.7257788
</td>
<td style="text-align:right;">
0.7561693
</td>
<td style="text-align:right;">
0.7527516
</td>
<td style="text-align:right;">
0.7261708
</td>
<td style="text-align:right;">
0.7797983
</td>
<td style="text-align:right;">
0.7607428
</td>
<td style="text-align:right;">
0.7370423
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.7734842
</td>
<td style="text-align:right;">
0.7515707
</td>
<td style="text-align:right;">
0.7243577
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_nordii_55557
</td>
<td style="text-align:right;">
0.0621894
</td>
<td style="text-align:right;">
0.0621223
</td>
<td style="text-align:right;">
0.0623676
</td>
<td style="text-align:right;">
0.0741022
</td>
<td style="text-align:right;">
0.3442077
</td>
<td style="text-align:right;">
0.0757429
</td>
<td style="text-align:right;">
0.8723801
</td>
<td style="text-align:right;">
0.9012884
</td>
<td style="text-align:right;">
0.8336514
</td>
<td style="text-align:right;">
0.8630350
</td>
<td style="text-align:right;">
0.8991728
</td>
<td style="text-align:right;">
0.8951088
</td>
<td style="text-align:right;">
0.8635012
</td>
<td style="text-align:right;">
0.9167779
</td>
<td style="text-align:right;">
0.8379055
</td>
<td style="text-align:right;">
0.8764286
</td>
<td style="text-align:right;">
0.7734842
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.8937046
</td>
<td style="text-align:right;">
0.8613451
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_ovatus_58035
</td>
<td style="text-align:right;">
0.0604276
</td>
<td style="text-align:right;">
0.0603623
</td>
<td style="text-align:right;">
0.0606006
</td>
<td style="text-align:right;">
0.0720028
</td>
<td style="text-align:right;">
0.3344560
</td>
<td style="text-align:right;">
0.0735971
</td>
<td style="text-align:right;">
0.9476879
</td>
<td style="text-align:right;">
0.9621451
</td>
<td style="text-align:right;">
0.8279024
</td>
<td style="text-align:right;">
0.8570834
</td>
<td style="text-align:right;">
0.9480576
</td>
<td style="text-align:right;">
0.9650359
</td>
<td style="text-align:right;">
0.8575464
</td>
<td style="text-align:right;">
0.8908048
</td>
<td style="text-align:right;">
0.8141668
</td>
<td style="text-align:right;">
0.8703847
</td>
<td style="text-align:right;">
0.7515707
</td>
<td style="text-align:right;">
0.8937046
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.8554052
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_rodentium_59708
</td>
<td style="text-align:right;">
0.0582396
</td>
<td style="text-align:right;">
0.0581767
</td>
<td style="text-align:right;">
0.0584064
</td>
<td style="text-align:right;">
0.0693957
</td>
<td style="text-align:right;">
0.3223459
</td>
<td style="text-align:right;">
0.0709323
</td>
<td style="text-align:right;">
0.8349946
</td>
<td style="text-align:right;">
0.8626640
</td>
<td style="text-align:right;">
0.8645899
</td>
<td style="text-align:right;">
0.9311080
</td>
<td style="text-align:right;">
0.8606391
</td>
<td style="text-align:right;">
0.8567492
</td>
<td style="text-align:right;">
0.9399769
</td>
<td style="text-align:right;">
0.8585503
</td>
<td style="text-align:right;">
0.7846873
</td>
<td style="text-align:right;">
0.9089547
</td>
<td style="text-align:right;">
0.7243577
</td>
<td style="text-align:right;">
0.8613451
</td>
<td style="text-align:right;">
0.8554052
</td>
<td style="text-align:right;">
1.0000000
</td>
</tr>
</tbody>
</table>
</div>
<p>Here is a heat map plot of the similarity matrix:
<img src="microbiome_tutorial_files/figure-html/unnamed-chunk-9-1.png" width="768"></p>
<p>Note that:</p>
<ul>
<li><p>The diagonal elements of this matrix are all 1, since each
species is identical to itself</p></li>
<li><p>The columns and rows are in the same order (i.e., column 1
corresponds to the same species as row 1, etc.)</p></li>
<li><p>The ordering of the species in the similarity matrix,
<code>xue_species_similarity</code>, matches the ordering in the
relative abundance matrix, <code>xue_microbiome_sample</code>.</p></li>
</ul>
</div>
<div class="section level3">
<h3 id="visualize-relative-abundances">1 - Visualize relative abundances<a class="anchor" aria-label="anchor" href="#visualize-relative-abundances"></a>
</h3>
<p>In order to visualize the community composition of each microbiome
sample from each subject, we generate a stacked bar plot using the
<code>plot_relabund</code> function from <em>FAVA</em>. Because
<code>plot_relabund</code> returns a <em>ggplot2</em> object, the
resulting plot can be customized using other functions from
<em>ggplot2</em>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Make a color palette for all 524 species</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">species_palette</span> <span class="op">=</span> <span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">turbo</a></span><span class="op">(</span><span class="fl">524</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">524</span><span class="op">)</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">`names&lt;-`</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">xue_microbiome_sample</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Make a ggplot2 stacked bar plot</span></span>
<span><span class="fu"><a href="../reference/plot_relabund.html">plot_relabund</a></span><span class="op">(</span><span class="va">xue_microbiome_sample</span>,</span>
<span>              group <span class="op">=</span> <span class="st">"subject"</span>,</span>
<span>              time <span class="op">=</span> <span class="st">"timepoint"</span>,</span>
<span>              arrange <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>              K <span class="op">=</span> <span class="fl">524</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="co"># Specify a custom color scheme</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">species_palette</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">species_palette</span><span class="op">)</span></span></code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-10-1.png" width="768"></p>
<p>Our example data set contains relative abundance samples from
multiple subjects that were taken at uneven time points. We account for
these properties by specifying the column name that describes which
group each sample belongs to (<code>group = "subject"</code>) as well as
which sampling day each sample corresponds to
(<code>time= "timepoint"</code>). Providing the <code>group</code>
parameter results in a plot that has one facet for each group. Providing
the <code>time</code> parameter results in a plot where each sample may
be repeated multiple times to reflect the number of days for which it
informs the composition. Since the sampling scheme (depicted above in <a href="#example-data">0 - Example data</a>) includes weekly samples at
the beginning and end of the study and daily samples in the middle, the
bars are narrower near the middle of each plot. Consider for reference
the plot below which does not have the <code>time</code> parameter
specified, and thus includes each sample exactly once.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_relabund.html">plot_relabund</a></span><span class="op">(</span><span class="va">xue_microbiome_sample</span>,</span>
<span>              group <span class="op">=</span> <span class="st">"subject"</span>,</span>
<span>              arrange <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>              K <span class="op">=</span> <span class="fl">524</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">species_palette</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">species_palette</span><span class="op">)</span></span></code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-11-1.png" width="768"></p>
<p>Consider also the plot below which specifies neither
<code>time</code> nor <code>group</code> and instead plots all samples
in a single plot.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_relabund.html">plot_relabund</a></span><span class="op">(</span><span class="va">xue_microbiome_sample</span>,</span>
<span>              arrange <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>              K <span class="op">=</span> <span class="fl">524</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">species_palette</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">species_palette</span><span class="op">)</span></span></code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-12-1.png" width="768"></p>
<p>In the above plots, we have specified
<code>arrange = "vertical"</code>, which vertically arranges species
from bottom to top in order of decreasing overall abundance. Specifying
<code>arrange="horizontal"</code> horizontally arranges samples from
left to right in order of increasing abundance of the most abundant
species. Specifying <code>arrange=TRUE</code> or
<code>arrange="both"</code> results in a plot with both types of
ordering. <code>arrange="both"</code> is a useful option for
highlighting patterns when the horizontal ordering of your samples does
not correspond to a meaningful property of the data, such as sampling
time.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_relabund.html">plot_relabund</a></span><span class="op">(</span><span class="va">xue_microbiome_sample</span>,</span>
<span>              arrange <span class="op">=</span> <span class="st">"both"</span>,</span>
<span>              K <span class="op">=</span> <span class="fl">524</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">species_palette</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">species_palette</span><span class="op">)</span></span></code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-13-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="compute-unweighted-fava">2 - Compute unweighted FAVA<a class="anchor" aria-label="anchor" href="#compute-unweighted-fava"></a>
</h3>
<p>The primary goal of the <em>FAVA</em> R package is to compute the
statistic FAVA, a measure of the variability across many relative
abundance vectors, introduced in the paper Morrison et al. This
statistic is computed using the function <code>fava</code>, which takes
a relative abundance matrix (<code>relab_matrix</code>) and computes the
variability across all the rows at once, returning a single index
between 0 and 1. If the relative abundance matrix contains metadata in
addition to relative abundances, the number of species, <code>K</code>,
must also be specified. If the matrix contains multiple groups we wish
to separately analyze, we must also specify the name of the matrix
column specifying group membership using the <code>group</code>
parameter. In our example, this column is called <code>"subject"</code>.
The code below computes FAVA across all samples from each subject.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/fava.html">fava</a></span><span class="op">(</span>relab_matrix <span class="op">=</span> <span class="va">xue_microbiome_sample</span>,</span>
<span>     group <span class="op">=</span> <span class="st">"subject"</span>,</span>
<span>     K <span class="op">=</span> <span class="fl">524</span><span class="op">)</span></span>
<span><span class="co">#&gt;   subject       FAVA</span></span>
<span><span class="co">#&gt; 1     XBA 0.15335493</span></span>
<span><span class="co">#&gt; 2     XDA 0.10891778</span></span>
<span><span class="co">#&gt; 3     XMA 0.09997127</span></span></code></pre></div>
<p>If <code>group</code> is not specified, FAVA is computed across all
samples in the matrix. For our example, this is a measure of the
variability across both time and subjects.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/fava.html">fava</a></span><span class="op">(</span>relab_matrix <span class="op">=</span> <span class="va">xue_microbiome_sample</span>,</span>
<span>     K <span class="op">=</span> <span class="fl">524</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.1666177</span></span></code></pre></div>
<p>If the number of samples/rows in your relative abundance matrix is
very small, you may wish to normalize FAVA by the theoretical upper
bound on FAVA conditional on the number of samples/rows and the
abundance of the most abundant taxon. Compute normalized FAVA by
specifying <code>normalized = TRUE</code> (see below). Refer to the
supplemental section <a href="#when-to-normalize-fava">When to normalize
FAVA</a> for more information.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/fava.html">fava</a></span><span class="op">(</span>relab_matrix <span class="op">=</span> <span class="va">xue_microbiome_sample</span>,</span>
<span>     group <span class="op">=</span> <span class="st">"subject"</span>,</span>
<span>     K <span class="op">=</span> <span class="fl">524</span>, </span>
<span>     normalized <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt;   subject      FAVA</span></span>
<span><span class="co">#&gt; 1     XBA 0.1564204</span></span>
<span><span class="co">#&gt; 2     XDA 0.1092958</span></span>
<span><span class="co">#&gt; 3     XMA 0.1012432</span></span></code></pre></div>
<div class="section level4">
<h4 id="compute-unweighted-fava-with-multiple-groups">Compute unweighted FAVA with multiple groups<a class="anchor" aria-label="anchor" href="#compute-unweighted-fava-with-multiple-groups"></a>
</h4>
<p>If the relative abundance matrix contains multiple categories I wish
to group by, a vector of column names can be provided as the
<code>group</code> parameter. For example, suppose I want to separately
compute FAVA across samples from each subject taken before and after the
antibiotic perturbation, which occurred on days 29 through 34.</p>
<p>First, I make a new column, called <code>Antibiotic</code>, that
states whether a sample is from before, during, or after the antibiotic
perturbation. It is important that this column is added the left of the
relative abundances.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">antibiotic_data</span> <span class="op">=</span> <span class="va">xue_microbiome_sample</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Antibiotic <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">timepoint</span> <span class="op">&lt;</span> <span class="fl">29</span>, <span class="st">"Before"</span>, </span>
<span>                             <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">timepoint</span> <span class="op">&lt;</span><span class="fl">35</span>, <span class="st">"During"</span>, <span class="st">"After"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         .after <span class="op">=</span> <span class="va">timepoint</span><span class="op">)</span></span></code></pre></div>
Here are the first 20 rows and 5 columns of this new matrix:
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:800px; ">
<table class="table">
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
subject
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
timepoint
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Antibiotic
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Actinomyces_sp_58647
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Actinomyces_sp_59161
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Before
</td>
<td style="text-align:right;">
3.00e-05
</td>
<td style="text-align:right;">
1.95e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
Before
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:left;">
Before
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:left;">
Before
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:left;">
Before
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:left;">
Before
</td>
<td style="text-align:right;">
4.30e-05
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:left;">
Before
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:left;">
Before
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:left;">
Before
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
28
</td>
<td style="text-align:left;">
Before
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:left;">
During
</td>
<td style="text-align:right;">
8.90e-06
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:left;">
During
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:left;">
After
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
3.16e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
37
</td>
<td style="text-align:left;">
After
</td>
<td style="text-align:right;">
2.52e-05
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
39
</td>
<td style="text-align:left;">
After
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
40
</td>
<td style="text-align:left;">
After
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
41
</td>
<td style="text-align:left;">
After
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
42
</td>
<td style="text-align:left;">
After
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
44
</td>
<td style="text-align:left;">
After
</td>
<td style="text-align:right;">
7.39e-05
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
XBA
</td>
<td style="text-align:right;">
45
</td>
<td style="text-align:left;">
After
</td>
<td style="text-align:right;">
1.61e-05
</td>
<td style="text-align:right;">
1.63e-05
</td>
</tr>
</tbody>
</table>
</div>
<p>Second, I remove samples from during the antibiotic perturbation:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">antibiotic_data</span> <span class="op">=</span> <span class="va">antibiotic_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Antibiotic</span> <span class="op">!=</span> <span class="st">"During"</span><span class="op">)</span></span></code></pre></div>
<p>Lastly, I compute FAVA specifying both <code>subject</code> and
<code>Antibiotic</code> as groups:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/fava.html">fava</a></span><span class="op">(</span>relab_matrix <span class="op">=</span> <span class="va">antibiotic_data</span>,</span>
<span>     group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"subject"</span>, <span class="st">"Antibiotic"</span><span class="op">)</span>,</span>
<span>     K <span class="op">=</span> <span class="fl">524</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   subject Antibiotic grouping_var_multiple   FAVA</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> XBA     Before     XBA_Before            0.014<span style="text-decoration: underline;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> XBA     After      XBA_After             0.094<span style="text-decoration: underline;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> XDA     Before     XDA_Before            0.018<span style="text-decoration: underline;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> XDA     After      XDA_After             0.031<span style="text-decoration: underline;">7</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> XMA     Before     XMA_Before            0.012<span style="text-decoration: underline;">8</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> XMA     After      XMA_After             0.031<span style="text-decoration: underline;">7</span></span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="compute-weighted-fava">3 - Compute weighted FAVA<a class="anchor" aria-label="anchor" href="#compute-weighted-fava"></a>
</h3>
<p>The statistic FAVA can be weighted in two possible ways:</p>
<ol style="list-style-type: decimal">
<li><p>Providing a species similarity matrix <code>S</code> allows FAVA
to account for the similarity among taxa.</p></li>
<li><p>Providing a weighting vector <code>w</code>, or the name of the
column corresponding to sampling times (which can be converted to a
weighting vector according to equations 5 and 6 of Morrison et al.),
allows FAVA to assign non-uniform weights to the samples.</p></li>
</ol>
<p><strong>(1)</strong> Here, we provide a phylogenetic similarity
matrix (<code>S = xue_species_similarity</code>) to FAVA so that its
computation can account for the varying levels of similarity between the
species in the data.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/fava.html">fava</a></span><span class="op">(</span>relab_matrix <span class="op">=</span> <span class="va">xue_microbiome_sample</span>,</span>
<span>     group <span class="op">=</span> <span class="st">"subject"</span>,</span>
<span>     K <span class="op">=</span> <span class="fl">524</span>,</span>
<span>     S <span class="op">=</span> <span class="va">xue_species_similarity</span><span class="op">)</span></span>
<span><span class="co">#&gt;   subject       FAVA</span></span>
<span><span class="co">#&gt; 1     XBA 0.11090489</span></span>
<span><span class="co">#&gt; 2     XDA 0.12424577</span></span>
<span><span class="co">#&gt; 3     XMA 0.05669292</span></span></code></pre></div>
<p><strong>(2)</strong> If the data set corresponds to time series data,
as our example does, providing the name of the matrix column that
specifies the time each sample was collected allows FAVA to compute a
weighting vector based on these sampling times and implement this
weighting vector in the computation. In
<code>xue_microbiome_sample</code>, the column is called “time
point.”</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/fava.html">fava</a></span><span class="op">(</span>relab_matrix <span class="op">=</span> <span class="va">xue_microbiome_sample</span>,</span>
<span>     group <span class="op">=</span> <span class="st">"subject"</span>,</span>
<span>     K <span class="op">=</span> <span class="fl">524</span>,</span>
<span>     time <span class="op">=</span> <span class="st">"timepoint"</span><span class="op">)</span></span>
<span><span class="co">#&gt;   subject       FAVA</span></span>
<span><span class="co">#&gt; 1     XBA 0.14494815</span></span>
<span><span class="co">#&gt; 2     XDA 0.10834448</span></span>
<span><span class="co">#&gt; 3     XMA 0.07088044</span></span></code></pre></div>
<p>An arbitrary weighting vector can instead be provided to the
<code>fava</code> function as the <code>w</code> parameter. You may
provide either <code>w</code> or <code>time</code> but not both.
<code>w</code> must have length equal to the number of rows in your data
set. If <code>w</code> provides the weights for one computation of FAVA
(i.e., a single group) its entries must sum to 1. If <code>w</code>
provides weights for multiple groups, each subset of <code>w</code>
corresponding to a single group must sum to 1. We can manually compute a
<code>w</code> vector from a vector of sampling times using the function
<code>time_weights</code> that is used by <code>fava</code> when
<code>time</code> is specified.</p>
<p>For example, consider only subject XMA. We first create a data frame,
<code>XMA</code>, containing only samples from subject XMA. We then
compute a weighting vector based on the times at which subject XMA
collected samples, <code>XBA$timepoint</code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">XMA</span> <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">xue_microbiome_sample</span>, <span class="va">subject</span> <span class="op">==</span> <span class="st">"XMA"</span><span class="op">)</span></span>
<span><span class="va">XMA</span><span class="op">$</span><span class="va">timepoint</span></span>
<span><span class="co">#&gt;  [1]  1  8 15 22 23 24 25 26 27 28 29 30 31 32 34 35 36 37 38 39 40 43 50 57 64</span></span>
<span></span>
<span><span class="va">weights</span> <span class="op">=</span> <span class="fu"><a href="../reference/time_weights.html">time_weights</a></span><span class="op">(</span>times <span class="op">=</span> <span class="va">XMA</span><span class="op">$</span><span class="va">timepoint</span><span class="op">)</span></span>
<span><span class="va">weights</span></span>
<span><span class="co">#&gt;  [1] 0.05555556 0.11111111 0.11111111 0.06349206 0.01587302 0.01587302</span></span>
<span><span class="co">#&gt;  [7] 0.01587302 0.01587302 0.01587302 0.01587302 0.01587302 0.01587302</span></span>
<span><span class="co">#&gt; [13] 0.01587302 0.02380952 0.02380952 0.01587302 0.01587302 0.01587302</span></span>
<span><span class="co">#&gt; [19] 0.01587302 0.01587302 0.03174603 0.07936508 0.11111111 0.11111111</span></span>
<span><span class="co">#&gt; [25] 0.05555556</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">weights</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1</span></span></code></pre></div>
<p>When we plot each sample’s weight based on when that sample was
collected, we see that the daily samples during the middle of the study
period have lower weights than the weekly samples during the beginning
and end of the study. We see that the subject missed a day of sampling
in the middle of the study, and the samples before and after this missed
day have slightly higher weights than the other daily samples.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">XMA</span><span class="op">$</span><span class="va">timepoint</span>,</span>
<span>                     y <span class="op">=</span> <span class="va">weights</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Study day"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Weight"</span><span class="op">)</span></span></code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-24-1.png" width="768"></p>
<p>We can use this weighting vector to compute the temporal variability
of subject XMA. Note that we get the same value of FAVA as when we
specified the <code>time</code> parameter above.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/fava.html">fava</a></span><span class="op">(</span>relab_matrix <span class="op">=</span> <span class="va">XMA</span>,</span>
<span>     K <span class="op">=</span> <span class="fl">524</span>,</span>
<span>     w <span class="op">=</span> <span class="va">weights</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.07088044</span></span>
<span></span>
<span><span class="fu"><a href="../reference/fava.html">fava</a></span><span class="op">(</span>relab_matrix <span class="op">=</span> <span class="va">XMA</span>,</span>
<span>     K <span class="op">=</span> <span class="fl">524</span>,</span>
<span>     time <span class="op">=</span> <span class="st">"timepoint"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.07088044</span></span></code></pre></div>
<p>You may incorporate both species similarity (specifying
<code>S</code>) and uneven row weightings (specifying <code>w</code> or
<code>time</code>) into the computation of FAVA.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fava_out</span> <span class="op">=</span> <span class="fu"><a href="../reference/fava.html">fava</a></span><span class="op">(</span>relab_matrix <span class="op">=</span> <span class="va">xue_microbiome_sample</span>,</span>
<span>     group <span class="op">=</span> <span class="st">"subject"</span>,</span>
<span>     K <span class="op">=</span> <span class="fl">524</span>,</span>
<span>     time <span class="op">=</span> <span class="st">"timepoint"</span>,</span>
<span>     S <span class="op">=</span> <span class="va">xue_species_similarity</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fava_out</span></span>
<span><span class="co">#&gt;   subject       FAVA</span></span>
<span><span class="co">#&gt; 1     XBA 0.11298430</span></span>
<span><span class="co">#&gt; 2     XDA 0.13109618</span></span>
<span><span class="co">#&gt; 3     XMA 0.04612452</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">fava_out</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">subject</span>, y <span class="op">=</span> <span class="va">FAVA</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-26-1.png" width="480"></p>
<p>We will specify both <code>S</code> and <code>time</code> in our
analysis of this data set because it contains many species, some very
similar and some very distantly related, and because the sampling scheme
is uneven, containing some weekly samples and some daily samples.</p>
</div>
<div class="section level3">
<h3 id="bootstrapping">4 - Bootstrapping<a class="anchor" aria-label="anchor" href="#bootstrapping"></a>
</h3>
<p>The preceding computation suggested that subject XDA had the most
temporally variable microbiome across the entire study period, while
subject XMA had much lower variability than either XDA or XBA. To
determine if these differences in variability levels are statistically
significant, we employ bootstrapping, a computational method used to
generate a null distribution against which to test statistical
hypotheses.</p>
<p>As an example, we will compare subject XDA, who has
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>n</mi><mo>=</mo><mn>24</mn></mrow><annotation encoding="application/x-tex">n=24</annotation></semantics></math>
samples, to subject XMA, with
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>m</mi><mo>=</mo><mn>25</mn></mrow><annotation encoding="application/x-tex">m=25</annotation></semantics></math>
samples. Bootstrapping proceeds in the following steps:</p>
<ol style="list-style-type: decimal">
<li><p>Merge the two relative abundance matrices to generate a single
matrix with
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>n</mi><mo>+</mo><mi>m</mi></mrow><annotation encoding="application/x-tex">n + m</annotation></semantics></math>
samples.</p></li>
<li><p>Draw
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>n</mi><annotation encoding="application/x-tex">n</annotation></semantics></math>
samples with replacement from this merged matrix. This generates a
bootstrap replicate of subject XDA. Compute FAVA on this
matrix.</p></li>
<li><p>Draw
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
samples with replacement from the merged matrix to generate a bootstrap
replicate of XMA. Compute FAVA on this matrix.</p></li>
<li><p>Compute the difference between the values of FAVA from steps 3
and 4.</p></li>
<li><p>Repeat steps 2-4 many times to generate many bootstrapped
differences in FAVA values between XDA and XMA under the null hypothesis
that there is no true difference in the distribution generating their
samples.</p></li>
<li><p>Compare the true difference in FAVA values between XDA and XMA to
the distribution of bootstrapped differences. The fraction of bootstrap
replicate differences whose absolute values are greater than the
absolute value of the true value is the two-sided P-value for our
statistical test. One-sided tests could be performed by computing the
fraction of differences greater or less than the true value.</p></li>
</ol>
<p>This procedure is implemented in the function
<code>bootstrap_fava</code>, which can conduct many pairwise
comparisons. In the below example, we generate 100 bootstrap replicates
(<code>n_replicates = 100</code>) of the difference in FAVA values
between each pair of subjects (<code>group = "subject"</code>) in our
relative abundance matrix
(<code>relab_matrix = xue_microbiome_sample</code>). We recommend using
500 or 1000 bootstrap replicates for a real analysis. We weight each
FAVA computation by the sampling times (<code>time = "timepoint"</code>)
and the species similarity (<code>S = xue_species_similarity</code>).
Because bootstrapping is a random process, running the code multiple
times would give slightly different results. Setting a random seed
(<code>seed = 3</code>) makes this result repeatable.
<!-- We specify `save_replicates = FALSE` so that `bootstrap_fava` does not save all 600 replicate matrices (100 per matrix in the 3 pairs of matrices being compared: XBA - XDA, XBA - XMA, and XDA - XMA), which would make the resulting object quite large.  -->
If you wish to perform a one-sided statistical test, you can specify
<code>alternative = "lesser"</code> or
<code>alternative = "greater"</code>; the default value is
<code>alternative = "two.sided"</code>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bootstrap_out</span> <span class="op">=</span> <span class="fu"><a href="../reference/bootstrap_fava.html">bootstrap_fava</a></span><span class="op">(</span>relab_matrix <span class="op">=</span> <span class="va">xue_microbiome_sample</span>,</span>
<span>                               n_replicates <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                               seed <span class="op">=</span> <span class="fl">3</span>,</span>
<span>                               group <span class="op">=</span> <span class="st">"subject"</span>,</span>
<span>                               K <span class="op">=</span> <span class="fl">524</span>,</span>
<span>                               time <span class="op">=</span> <span class="st">"timepoint"</span>,</span>
<span>                               S <span class="op">=</span> <span class="va">xue_species_similarity</span><span class="op">)</span></span></code></pre></div>
<p>The resulting list, which we have here named
<code>bootstrap_out</code>, includes a matrix of p-values for each
pairwise comparison (<code>P_values</code>) and a plot
(<code>bootstrap_distribution_plot</code>) showing for each pair of
matrices the bootstrap distribution of differences in FAVA values (black
dots) compared to the true difference (red dot).</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">bootstrap_out</span>, max.level <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 4</span></span>
<span><span class="co">#&gt;  $ P_values                   :'data.frame': 3 obs. of  3 variables:</span></span>
<span><span class="co">#&gt;  $ bootstrap_distribution_plot:List of 11</span></span>
<span><span class="co">#&gt;   ..- attr(*, "class")= chr [1:2] "gg" "ggplot"</span></span>
<span><span class="co">#&gt;  $ observed_difference        :'data.frame': 3 obs. of  2 variables:</span></span>
<span><span class="co">#&gt;  $ bootstrap_difference       : tibble [300 × 3] (S3: tbl_df/tbl/data.frame)</span></span>
<span></span>
<span><span class="va">bootstrap_out</span><span class="op">$</span><span class="va">P_values</span></span>
<span><span class="co">#&gt;   Comparison P_value P_value_numeric</span></span>
<span><span class="co">#&gt; 1  XBA - XDA    0.56            0.56</span></span>
<span><span class="co">#&gt; 2  XBA - XMA    0.14            0.14</span></span>
<span><span class="co">#&gt; 3  XDA - XMA    0.04            0.04</span></span></code></pre></div>
<p>Our bootstrap statistical test fails to find a significant difference
between the FAVA values of subjects XBA and XDA (one-sided
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>=</mo><mn>0.6</mn></mrow><annotation encoding="application/x-tex">P=0.6</annotation></semantics></math>).
However, there are significant differences in FAVA values between XBA
and XMA (one-sided
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>=</mo><mn>0.08</mn></mrow><annotation encoding="application/x-tex">P=0.08</annotation></semantics></math>)
and between XDA and XMA (one-sided
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>&lt;</mo><mn>0.01</mn></mrow><annotation encoding="application/x-tex">P&lt;0.01</annotation></semantics></math>,
which suggests that none of the 100 replicate differences were greater
than the true difference). We can visualize how the bootstrap
distribution of pairwise differences in FAVA values compares to the true
differences for each pair in the following plot:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bootstrap_out</span><span class="op">$</span><span class="va">bootstrap_distribution_plot</span></span></code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-29-1.png" width="768"></p>
<p>We can see that the true difference in FAVA between XBA and XDA (red
dot, left panel) falls near the center of the bootstrap distribution,
suggesting the observed difference is not unlikely under the null model
that there are no differences between XBA and XDA. However, the other
two panels show observed FAVA differences (red dots) further from the
center of the distribution, suggesting that there are true differences
between those pairs of subjects.</p>
</div>
<div class="section level3">
<h3 id="sliding-windows">5 - Sliding windows<a class="anchor" aria-label="anchor" href="#sliding-windows"></a>
</h3>
<p>Finally, in order to explore how temporal variability changes over
the course of the study period for each subject, we compute FAVA in
sliding windows. We here use sliding windows 6-samples wide separated by
1-sample increments, but those values can be customized using the
<code>window_size</code> and <code>window_step</code> parameters. Like
<code>bootstrap_fava</code>, the function <code>window_fava</code>
returns a list of objects. <code>window_fava$window_data</code> is a
data frame containing the value of FAVA for each sliding window.
<code>window_fava$window_plot</code> plots the value of FAVA for each
window as a horizontal line segment with length corresponding to the
samples included and vertical position determined by the window’s FAVA
value.</p>
<div class="section level4">
<h4 id="compute-fava-for-each-window">Compute FAVA for each window<a class="anchor" aria-label="anchor" href="#compute-fava-for-each-window"></a>
</h4>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">window_out</span> <span class="op">=</span> <span class="fu"><a href="../reference/window_fava.html">window_fava</a></span><span class="op">(</span>relab_matrix <span class="op">=</span> <span class="va">xue_microbiome_sample</span>,</span>
<span>                         window_size <span class="op">=</span> <span class="fl">6</span>, window_step <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                         K <span class="op">=</span> <span class="fl">524</span>,</span>
<span>                         time <span class="op">=</span> <span class="st">"timepoint"</span>,</span>
<span>                         S <span class="op">=</span> <span class="va">xue_species_similarity</span>,</span>
<span>                         group <span class="op">=</span> <span class="st">"subject"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">window_out</span><span class="op">$</span><span class="va">window_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 9</span></span></span>
<span><span class="co">#&gt;   group    FAVA window_index    w1    w2    w3    w4    w5    w6</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> XBA   0.012<span style="text-decoration: underline;">8</span>             1     1     8    15    22    23    24</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> XBA   0.011<span style="text-decoration: underline;">7</span>             2     8    15    22    23    24    25</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> XBA   0.012<span style="text-decoration: underline;">7</span>             3    15    22    23    24    25    26</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> XBA   0.009<span style="text-decoration: underline;">98</span>            4    22    23    24    25    26    27</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> XBA   0.009<span style="text-decoration: underline;">13</span>            5    23    24    25    26    27    28</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> XBA   0.011<span style="text-decoration: underline;">4</span>             6    24    25    26    27    28    29</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="visualize-fava-in-sliding-windows">Visualize FAVA in sliding windows<a class="anchor" aria-label="anchor" href="#visualize-fava-in-sliding-windows"></a>
</h4>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">window_out</span><span class="op">$</span><span class="va">window_plot</span></span></code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-31-1.png" width="768"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">window_out</span><span class="op">$</span><span class="va">window_plot</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">group</span><span class="op">)</span></span></code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-31-2.png" width="768"></p>
<p>We see that each subject experiences an increase in variability when
they are taking the antibiotic. Both subjects XDA and XMA return to FAVA
values similar to their pre-antibiotic values, but subject XBA does not
re-stabilize during the study period.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>This concludes the tutorial on the application of the <em>FAVA</em> R
package to analysis of microbiome data. We hope you found it helpful!
For guidance on specific functions, access the documentation by typing
<code>?</code> into the R console (e.g., <code><a href="../reference/fava.html">?fava</a></code>). To see all
available functions, type <code>?FAVA::</code>. For more details on the
FAVA statistic, see the paper by Morrison et al.</p>
</div>
<div class="section level2">
<h2 id="supplemental-materials">Supplemental materials<a class="anchor" aria-label="anchor" href="#supplemental-materials"></a>
</h2>
<div class="section level3">
<h3 id="generating-a-similarity-matrix">Generating a similarity matrix<a class="anchor" aria-label="anchor" href="#generating-a-similarity-matrix"></a>
</h3>
<p>In this section, we provide more detail on how to generate a
similarity matrix from a phylogenetic tree. We proceed in three
steps:</p>
<ol style="list-style-type: decimal">
<li><p>Convert a phylogenetic tree to a pairwise phylogenetic distance
matrix</p></li>
<li><p>Explore some of the different transformations that can be used to
convert a distance matrix to a similarity matrix</p></li>
<li><p>Explore how our choice of transformation influences the results
of FAVA for our example data</p></li>
</ol>
<div class="section level4">
<h4 id="convert-a-phylogenetic-tree-to-a-pairwise-phylogenetic-distance-matrix">(1) Convert a phylogenetic tree to a pairwise phylogenetic distance
matrix<a class="anchor" aria-label="anchor" href="#convert-a-phylogenetic-tree-to-a-pairwise-phylogenetic-distance-matrix"></a>
</h4>
<p>We begin with a phylogenetic tree, <code>xue_species_tree</code>,
which describes the evolutionary relationships among the species based
on their sequence data. In R, a tree is stored as a list describing the
nodes, edges, and tips of the phylogenetic tree:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">xue_species_tree</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 5</span></span>
<span><span class="co">#&gt;  $ edge       : int [1:1046, 1:2] 525 525 526 526 527 528 528 529 530 530 ...</span></span>
<span><span class="co">#&gt;  $ edge.length: num [1:1046] 0.00301 0.00115 0.00654 0.00085 0.00099 ...</span></span>
<span><span class="co">#&gt;  $ Nnode      : int 523</span></span>
<span><span class="co">#&gt;  $ node.label : chr [1:523] "OROOT" "0.578" "0.253" "0.865" ...</span></span>
<span><span class="co">#&gt;  $ tip.label  : chr [1:524] "Enterobacter_cloacae_57303" "Enterobacter_cloacae_55011" "Enterobacter_cloacae_60571" "Enterobacter_cancerogenus_61658" ...</span></span>
<span><span class="co">#&gt;  - attr(*, "class")= chr "phylo"</span></span>
<span><span class="co">#&gt;  - attr(*, "order")= chr "cladewise"</span></span></code></pre></div>
<p>The R package <code>ape</code> contains many functions for
manipulating phylogenetic trees. For example, we visualize the tree with
the function <code>plot.phylo</code>.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/plot.phylo.html" class="external-link">plot.phylo</a></span><span class="op">(</span><span class="va">xue_species_tree</span>, cex <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span></code></pre></div>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-33-1.png" width="768"></p>
<p>We transform the tree into a distance matrix using the function
<code>cophenetic.phylo</code>, which computes the distance between a
pair of species using the branch lengths of the phylogenetic tree.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">distance_matrix</span> <span class="op">=</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/cophenetic.phylo.html" class="external-link">cophenetic.phylo</a></span><span class="op">(</span><span class="va">xue_species_tree</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">distance_matrix</span><span class="op">)</span></span>
<span><span class="co">#&gt;  num [1:524, 1:524] 0 0.0107 0.0157 0.0247 0.0226 ...</span></span>
<span><span class="co">#&gt;  - attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">#&gt;   ..$ : chr [1:524] "Enterobacter_cloacae_57303" "Enterobacter_cloacae_55011" "Enterobacter_cloacae_60571" "Enterobacter_cancerogenus_61658" ...</span></span>
<span><span class="co">#&gt;   ..$ : chr [1:524] "Enterobacter_cloacae_57303" "Enterobacter_cloacae_55011" "Enterobacter_cloacae_60571" "Enterobacter_cancerogenus_61658" ...</span></span></code></pre></div>
<p>Before moving on, we need to make the rows and columns of
<code>distance_matrix</code> match those of the relative abundance
matrix <code>xue_microbiome_sample</code>.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">species_order</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">xue_microbiome_sample</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">distance_matrix</span> <span class="op">=</span> <span class="va">distance_matrix</span><span class="op">[</span><span class="va">species_order</span>, <span class="va">species_order</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">distance_matrix</span><span class="op">)</span></span>
<span><span class="co">#&gt;  num [1:524, 1:524] 0 0.00756 0.08838 2.71274 2.88126 ...</span></span>
<span><span class="co">#&gt;  - attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">#&gt;   ..$ : chr [1:524] "Actinomyces_sp_58647" "Actinomyces_sp_59161" "Actinomyces_sp_59944" "Aggregatibacter_aphrophilus_58143" ...</span></span>
<span><span class="co">#&gt;   ..$ : chr [1:524] "Actinomyces_sp_58647" "Actinomyces_sp_59161" "Actinomyces_sp_59944" "Aggregatibacter_aphrophilus_58143" ...</span></span></code></pre></div>
Below are the first 40 rows and the first 20 columns of this distance
matrix, <code>distance_matrix</code>. Note that the diagonal elements
are all 0, since each species has distance 0 from itself.
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:800px; ">
<table class="table">
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Actinomyces_sp_58647
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Actinomyces_sp_59161
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Actinomyces_sp_59944
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Aggregatibacter_aphrophilus_58143
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Alistipes_finegoldii_56071
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Anaerostipes_hadrus_55206
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_acidifaciens_59693
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_caccae_53434
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_cellulosilyticus_58046
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_clarus_62282
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_faecis_58503
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_finegoldii_57739
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_fluxus_62283
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_fragilis_54507
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_graminisolvens_48736
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_intestinalis_61596
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_massiliensis_44749
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_nordii_55557
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_ovatus_58035
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Bacteroides_rodentium_59708
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Actinomyces_sp_58647
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00756
</td>
<td style="text-align:right;">
0.08838
</td>
<td style="text-align:right;">
2.71274
</td>
<td style="text-align:right;">
2.88126
</td>
<td style="text-align:right;">
2.56276
</td>
<td style="text-align:right;">
2.83046
</td>
<td style="text-align:right;">
2.79786
</td>
<td style="text-align:right;">
2.87587
</td>
<td style="text-align:right;">
2.84123
</td>
<td style="text-align:right;">
2.80021
</td>
<td style="text-align:right;">
2.80474
</td>
<td style="text-align:right;">
2.84069
</td>
<td style="text-align:right;">
2.76944
</td>
<td style="text-align:right;">
2.79418
</td>
<td style="text-align:right;">
2.82583
</td>
<td style="text-align:right;">
2.74660
</td>
<td style="text-align:right;">
2.77757
</td>
<td style="text-align:right;">
2.80631
</td>
<td style="text-align:right;">
2.84319
</td>
</tr>
<tr>
<td style="text-align:left;">
Actinomyces_sp_59161
</td>
<td style="text-align:right;">
0.00756
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.08946
</td>
<td style="text-align:right;">
2.71382
</td>
<td style="text-align:right;">
2.88234
</td>
<td style="text-align:right;">
2.56384
</td>
<td style="text-align:right;">
2.83154
</td>
<td style="text-align:right;">
2.79894
</td>
<td style="text-align:right;">
2.87695
</td>
<td style="text-align:right;">
2.84231
</td>
<td style="text-align:right;">
2.80129
</td>
<td style="text-align:right;">
2.80582
</td>
<td style="text-align:right;">
2.84177
</td>
<td style="text-align:right;">
2.77052
</td>
<td style="text-align:right;">
2.79526
</td>
<td style="text-align:right;">
2.82691
</td>
<td style="text-align:right;">
2.74768
</td>
<td style="text-align:right;">
2.77865
</td>
<td style="text-align:right;">
2.80739
</td>
<td style="text-align:right;">
2.84427
</td>
</tr>
<tr>
<td style="text-align:left;">
Actinomyces_sp_59944
</td>
<td style="text-align:right;">
0.08838
</td>
<td style="text-align:right;">
0.08946
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
2.70988
</td>
<td style="text-align:right;">
2.87840
</td>
<td style="text-align:right;">
2.55990
</td>
<td style="text-align:right;">
2.82760
</td>
<td style="text-align:right;">
2.79500
</td>
<td style="text-align:right;">
2.87301
</td>
<td style="text-align:right;">
2.83837
</td>
<td style="text-align:right;">
2.79735
</td>
<td style="text-align:right;">
2.80188
</td>
<td style="text-align:right;">
2.83783
</td>
<td style="text-align:right;">
2.76658
</td>
<td style="text-align:right;">
2.79132
</td>
<td style="text-align:right;">
2.82297
</td>
<td style="text-align:right;">
2.74374
</td>
<td style="text-align:right;">
2.77471
</td>
<td style="text-align:right;">
2.80345
</td>
<td style="text-align:right;">
2.84033
</td>
</tr>
<tr>
<td style="text-align:left;">
Aggregatibacter_aphrophilus_58143
</td>
<td style="text-align:right;">
2.71274
</td>
<td style="text-align:right;">
2.71382
</td>
<td style="text-align:right;">
2.70988
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
2.70600
</td>
<td style="text-align:right;">
2.51558
</td>
<td style="text-align:right;">
2.65520
</td>
<td style="text-align:right;">
2.62260
</td>
<td style="text-align:right;">
2.70061
</td>
<td style="text-align:right;">
2.66597
</td>
<td style="text-align:right;">
2.62495
</td>
<td style="text-align:right;">
2.62948
</td>
<td style="text-align:right;">
2.66543
</td>
<td style="text-align:right;">
2.59418
</td>
<td style="text-align:right;">
2.61892
</td>
<td style="text-align:right;">
2.65057
</td>
<td style="text-align:right;">
2.57134
</td>
<td style="text-align:right;">
2.60231
</td>
<td style="text-align:right;">
2.63105
</td>
<td style="text-align:right;">
2.66793
</td>
</tr>
<tr>
<td style="text-align:left;">
Alistipes_finegoldii_56071
</td>
<td style="text-align:right;">
2.88126
</td>
<td style="text-align:right;">
2.88234
</td>
<td style="text-align:right;">
2.87840
</td>
<td style="text-align:right;">
2.70600
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
2.68410
</td>
<td style="text-align:right;">
1.11940
</td>
<td style="text-align:right;">
1.08680
</td>
<td style="text-align:right;">
1.16481
</td>
<td style="text-align:right;">
1.13017
</td>
<td style="text-align:right;">
1.08915
</td>
<td style="text-align:right;">
1.09368
</td>
<td style="text-align:right;">
1.12963
</td>
<td style="text-align:right;">
1.05838
</td>
<td style="text-align:right;">
1.08312
</td>
<td style="text-align:right;">
1.11477
</td>
<td style="text-align:right;">
1.03554
</td>
<td style="text-align:right;">
1.06651
</td>
<td style="text-align:right;">
1.09525
</td>
<td style="text-align:right;">
1.13213
</td>
</tr>
<tr>
<td style="text-align:left;">
Anaerostipes_hadrus_55206
</td>
<td style="text-align:right;">
2.56276
</td>
<td style="text-align:right;">
2.56384
</td>
<td style="text-align:right;">
2.55990
</td>
<td style="text-align:right;">
2.51558
</td>
<td style="text-align:right;">
2.68410
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
2.63330
</td>
<td style="text-align:right;">
2.60070
</td>
<td style="text-align:right;">
2.67871
</td>
<td style="text-align:right;">
2.64407
</td>
<td style="text-align:right;">
2.60305
</td>
<td style="text-align:right;">
2.60758
</td>
<td style="text-align:right;">
2.64353
</td>
<td style="text-align:right;">
2.57228
</td>
<td style="text-align:right;">
2.59702
</td>
<td style="text-align:right;">
2.62867
</td>
<td style="text-align:right;">
2.54944
</td>
<td style="text-align:right;">
2.58041
</td>
<td style="text-align:right;">
2.60915
</td>
<td style="text-align:right;">
2.64603
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_acidifaciens_59693
</td>
<td style="text-align:right;">
2.83046
</td>
<td style="text-align:right;">
2.83154
</td>
<td style="text-align:right;">
2.82760
</td>
<td style="text-align:right;">
2.65520
</td>
<td style="text-align:right;">
1.11940
</td>
<td style="text-align:right;">
2.63330
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.06274
</td>
<td style="text-align:right;">
0.21301
</td>
<td style="text-align:right;">
0.17837
</td>
<td style="text-align:right;">
0.07749
</td>
<td style="text-align:right;">
0.05974
</td>
<td style="text-align:right;">
0.17783
</td>
<td style="text-align:right;">
0.13978
</td>
<td style="text-align:right;">
0.22974
</td>
<td style="text-align:right;">
0.16297
</td>
<td style="text-align:right;">
0.30974
</td>
<td style="text-align:right;">
0.13653
</td>
<td style="text-align:right;">
0.05373
</td>
<td style="text-align:right;">
0.18033
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_caccae_53434
</td>
<td style="text-align:right;">
2.79786
</td>
<td style="text-align:right;">
2.79894
</td>
<td style="text-align:right;">
2.79500
</td>
<td style="text-align:right;">
2.62260
</td>
<td style="text-align:right;">
1.08680
</td>
<td style="text-align:right;">
2.60070
</td>
<td style="text-align:right;">
0.06274
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.18041
</td>
<td style="text-align:right;">
0.14577
</td>
<td style="text-align:right;">
0.04489
</td>
<td style="text-align:right;">
0.03702
</td>
<td style="text-align:right;">
0.14523
</td>
<td style="text-align:right;">
0.10718
</td>
<td style="text-align:right;">
0.19714
</td>
<td style="text-align:right;">
0.13037
</td>
<td style="text-align:right;">
0.27714
</td>
<td style="text-align:right;">
0.10393
</td>
<td style="text-align:right;">
0.03859
</td>
<td style="text-align:right;">
0.14773
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_cellulosilyticus_58046
</td>
<td style="text-align:right;">
2.87587
</td>
<td style="text-align:right;">
2.87695
</td>
<td style="text-align:right;">
2.87301
</td>
<td style="text-align:right;">
2.70061
</td>
<td style="text-align:right;">
1.16481
</td>
<td style="text-align:right;">
2.67871
</td>
<td style="text-align:right;">
0.21301
</td>
<td style="text-align:right;">
0.18041
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.14354
</td>
<td style="text-align:right;">
0.18276
</td>
<td style="text-align:right;">
0.18729
</td>
<td style="text-align:right;">
0.14300
</td>
<td style="text-align:right;">
0.18519
</td>
<td style="text-align:right;">
0.27515
</td>
<td style="text-align:right;">
0.06704
</td>
<td style="text-align:right;">
0.35515
</td>
<td style="text-align:right;">
0.18194
</td>
<td style="text-align:right;">
0.18886
</td>
<td style="text-align:right;">
0.14550
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_clarus_62282
</td>
<td style="text-align:right;">
2.84123
</td>
<td style="text-align:right;">
2.84231
</td>
<td style="text-align:right;">
2.83837
</td>
<td style="text-align:right;">
2.66597
</td>
<td style="text-align:right;">
1.13017
</td>
<td style="text-align:right;">
2.64407
</td>
<td style="text-align:right;">
0.17837
</td>
<td style="text-align:right;">
0.14577
</td>
<td style="text-align:right;">
0.14354
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.14812
</td>
<td style="text-align:right;">
0.15265
</td>
<td style="text-align:right;">
0.06888
</td>
<td style="text-align:right;">
0.15055
</td>
<td style="text-align:right;">
0.24051
</td>
<td style="text-align:right;">
0.09350
</td>
<td style="text-align:right;">
0.32051
</td>
<td style="text-align:right;">
0.14730
</td>
<td style="text-align:right;">
0.15422
</td>
<td style="text-align:right;">
0.07138
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_faecis_58503
</td>
<td style="text-align:right;">
2.80021
</td>
<td style="text-align:right;">
2.80129
</td>
<td style="text-align:right;">
2.79735
</td>
<td style="text-align:right;">
2.62495
</td>
<td style="text-align:right;">
1.08915
</td>
<td style="text-align:right;">
2.60305
</td>
<td style="text-align:right;">
0.07749
</td>
<td style="text-align:right;">
0.04489
</td>
<td style="text-align:right;">
0.18276
</td>
<td style="text-align:right;">
0.14812
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.05177
</td>
<td style="text-align:right;">
0.14758
</td>
<td style="text-align:right;">
0.10953
</td>
<td style="text-align:right;">
0.19949
</td>
<td style="text-align:right;">
0.13272
</td>
<td style="text-align:right;">
0.27949
</td>
<td style="text-align:right;">
0.10628
</td>
<td style="text-align:right;">
0.05334
</td>
<td style="text-align:right;">
0.15008
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_finegoldii_57739
</td>
<td style="text-align:right;">
2.80474
</td>
<td style="text-align:right;">
2.80582
</td>
<td style="text-align:right;">
2.80188
</td>
<td style="text-align:right;">
2.62948
</td>
<td style="text-align:right;">
1.09368
</td>
<td style="text-align:right;">
2.60758
</td>
<td style="text-align:right;">
0.05974
</td>
<td style="text-align:right;">
0.03702
</td>
<td style="text-align:right;">
0.18729
</td>
<td style="text-align:right;">
0.15265
</td>
<td style="text-align:right;">
0.05177
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.15211
</td>
<td style="text-align:right;">
0.11406
</td>
<td style="text-align:right;">
0.20402
</td>
<td style="text-align:right;">
0.13725
</td>
<td style="text-align:right;">
0.28402
</td>
<td style="text-align:right;">
0.11081
</td>
<td style="text-align:right;">
0.03559
</td>
<td style="text-align:right;">
0.15461
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_fluxus_62283
</td>
<td style="text-align:right;">
2.84069
</td>
<td style="text-align:right;">
2.84177
</td>
<td style="text-align:right;">
2.83783
</td>
<td style="text-align:right;">
2.66543
</td>
<td style="text-align:right;">
1.12963
</td>
<td style="text-align:right;">
2.64353
</td>
<td style="text-align:right;">
0.17783
</td>
<td style="text-align:right;">
0.14523
</td>
<td style="text-align:right;">
0.14300
</td>
<td style="text-align:right;">
0.06888
</td>
<td style="text-align:right;">
0.14758
</td>
<td style="text-align:right;">
0.15211
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.15001
</td>
<td style="text-align:right;">
0.23997
</td>
<td style="text-align:right;">
0.09296
</td>
<td style="text-align:right;">
0.31997
</td>
<td style="text-align:right;">
0.14676
</td>
<td style="text-align:right;">
0.15368
</td>
<td style="text-align:right;">
0.06190
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_fragilis_54507
</td>
<td style="text-align:right;">
2.76944
</td>
<td style="text-align:right;">
2.77052
</td>
<td style="text-align:right;">
2.76658
</td>
<td style="text-align:right;">
2.59418
</td>
<td style="text-align:right;">
1.05838
</td>
<td style="text-align:right;">
2.57228
</td>
<td style="text-align:right;">
0.13978
</td>
<td style="text-align:right;">
0.10718
</td>
<td style="text-align:right;">
0.18519
</td>
<td style="text-align:right;">
0.15055
</td>
<td style="text-align:right;">
0.10953
</td>
<td style="text-align:right;">
0.11406
</td>
<td style="text-align:right;">
0.15001
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.16872
</td>
<td style="text-align:right;">
0.13515
</td>
<td style="text-align:right;">
0.24872
</td>
<td style="text-align:right;">
0.08689
</td>
<td style="text-align:right;">
0.11563
</td>
<td style="text-align:right;">
0.15251
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_graminisolvens_48736
</td>
<td style="text-align:right;">
2.79418
</td>
<td style="text-align:right;">
2.79526
</td>
<td style="text-align:right;">
2.79132
</td>
<td style="text-align:right;">
2.61892
</td>
<td style="text-align:right;">
1.08312
</td>
<td style="text-align:right;">
2.59702
</td>
<td style="text-align:right;">
0.22974
</td>
<td style="text-align:right;">
0.19714
</td>
<td style="text-align:right;">
0.27515
</td>
<td style="text-align:right;">
0.24051
</td>
<td style="text-align:right;">
0.19949
</td>
<td style="text-align:right;">
0.20402
</td>
<td style="text-align:right;">
0.23997
</td>
<td style="text-align:right;">
0.16872
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.22511
</td>
<td style="text-align:right;">
0.27346
</td>
<td style="text-align:right;">
0.17685
</td>
<td style="text-align:right;">
0.20559
</td>
<td style="text-align:right;">
0.24247
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_intestinalis_61596
</td>
<td style="text-align:right;">
2.82583
</td>
<td style="text-align:right;">
2.82691
</td>
<td style="text-align:right;">
2.82297
</td>
<td style="text-align:right;">
2.65057
</td>
<td style="text-align:right;">
1.11477
</td>
<td style="text-align:right;">
2.62867
</td>
<td style="text-align:right;">
0.16297
</td>
<td style="text-align:right;">
0.13037
</td>
<td style="text-align:right;">
0.06704
</td>
<td style="text-align:right;">
0.09350
</td>
<td style="text-align:right;">
0.13272
</td>
<td style="text-align:right;">
0.13725
</td>
<td style="text-align:right;">
0.09296
</td>
<td style="text-align:right;">
0.13515
</td>
<td style="text-align:right;">
0.22511
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.30511
</td>
<td style="text-align:right;">
0.13190
</td>
<td style="text-align:right;">
0.13882
</td>
<td style="text-align:right;">
0.09546
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_massiliensis_44749
</td>
<td style="text-align:right;">
2.74660
</td>
<td style="text-align:right;">
2.74768
</td>
<td style="text-align:right;">
2.74374
</td>
<td style="text-align:right;">
2.57134
</td>
<td style="text-align:right;">
1.03554
</td>
<td style="text-align:right;">
2.54944
</td>
<td style="text-align:right;">
0.30974
</td>
<td style="text-align:right;">
0.27714
</td>
<td style="text-align:right;">
0.35515
</td>
<td style="text-align:right;">
0.32051
</td>
<td style="text-align:right;">
0.27949
</td>
<td style="text-align:right;">
0.28402
</td>
<td style="text-align:right;">
0.31997
</td>
<td style="text-align:right;">
0.24872
</td>
<td style="text-align:right;">
0.27346
</td>
<td style="text-align:right;">
0.30511
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.25685
</td>
<td style="text-align:right;">
0.28559
</td>
<td style="text-align:right;">
0.32247
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_nordii_55557
</td>
<td style="text-align:right;">
2.77757
</td>
<td style="text-align:right;">
2.77865
</td>
<td style="text-align:right;">
2.77471
</td>
<td style="text-align:right;">
2.60231
</td>
<td style="text-align:right;">
1.06651
</td>
<td style="text-align:right;">
2.58041
</td>
<td style="text-align:right;">
0.13653
</td>
<td style="text-align:right;">
0.10393
</td>
<td style="text-align:right;">
0.18194
</td>
<td style="text-align:right;">
0.14730
</td>
<td style="text-align:right;">
0.10628
</td>
<td style="text-align:right;">
0.11081
</td>
<td style="text-align:right;">
0.14676
</td>
<td style="text-align:right;">
0.08689
</td>
<td style="text-align:right;">
0.17685
</td>
<td style="text-align:right;">
0.13190
</td>
<td style="text-align:right;">
0.25685
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.11238
</td>
<td style="text-align:right;">
0.14926
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_ovatus_58035
</td>
<td style="text-align:right;">
2.80631
</td>
<td style="text-align:right;">
2.80739
</td>
<td style="text-align:right;">
2.80345
</td>
<td style="text-align:right;">
2.63105
</td>
<td style="text-align:right;">
1.09525
</td>
<td style="text-align:right;">
2.60915
</td>
<td style="text-align:right;">
0.05373
</td>
<td style="text-align:right;">
0.03859
</td>
<td style="text-align:right;">
0.18886
</td>
<td style="text-align:right;">
0.15422
</td>
<td style="text-align:right;">
0.05334
</td>
<td style="text-align:right;">
0.03559
</td>
<td style="text-align:right;">
0.15368
</td>
<td style="text-align:right;">
0.11563
</td>
<td style="text-align:right;">
0.20559
</td>
<td style="text-align:right;">
0.13882
</td>
<td style="text-align:right;">
0.28559
</td>
<td style="text-align:right;">
0.11238
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.15618
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_rodentium_59708
</td>
<td style="text-align:right;">
2.84319
</td>
<td style="text-align:right;">
2.84427
</td>
<td style="text-align:right;">
2.84033
</td>
<td style="text-align:right;">
2.66793
</td>
<td style="text-align:right;">
1.13213
</td>
<td style="text-align:right;">
2.64603
</td>
<td style="text-align:right;">
0.18033
</td>
<td style="text-align:right;">
0.14773
</td>
<td style="text-align:right;">
0.14550
</td>
<td style="text-align:right;">
0.07138
</td>
<td style="text-align:right;">
0.15008
</td>
<td style="text-align:right;">
0.15461
</td>
<td style="text-align:right;">
0.06190
</td>
<td style="text-align:right;">
0.15251
</td>
<td style="text-align:right;">
0.24247
</td>
<td style="text-align:right;">
0.09546
</td>
<td style="text-align:right;">
0.32247
</td>
<td style="text-align:right;">
0.14926
</td>
<td style="text-align:right;">
0.15618
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_sartorii_54642
</td>
<td style="text-align:right;">
2.76142
</td>
<td style="text-align:right;">
2.76250
</td>
<td style="text-align:right;">
2.75856
</td>
<td style="text-align:right;">
2.58616
</td>
<td style="text-align:right;">
1.05036
</td>
<td style="text-align:right;">
2.56426
</td>
<td style="text-align:right;">
0.32456
</td>
<td style="text-align:right;">
0.29196
</td>
<td style="text-align:right;">
0.36997
</td>
<td style="text-align:right;">
0.33533
</td>
<td style="text-align:right;">
0.29431
</td>
<td style="text-align:right;">
0.29884
</td>
<td style="text-align:right;">
0.33479
</td>
<td style="text-align:right;">
0.26354
</td>
<td style="text-align:right;">
0.28828
</td>
<td style="text-align:right;">
0.31993
</td>
<td style="text-align:right;">
0.08834
</td>
<td style="text-align:right;">
0.27167
</td>
<td style="text-align:right;">
0.30041
</td>
<td style="text-align:right;">
0.33729
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_uniformis_57318
</td>
<td style="text-align:right;">
2.91147
</td>
<td style="text-align:right;">
2.91255
</td>
<td style="text-align:right;">
2.90861
</td>
<td style="text-align:right;">
2.73621
</td>
<td style="text-align:right;">
1.20041
</td>
<td style="text-align:right;">
2.71431
</td>
<td style="text-align:right;">
0.24861
</td>
<td style="text-align:right;">
0.21601
</td>
<td style="text-align:right;">
0.21378
</td>
<td style="text-align:right;">
0.13966
</td>
<td style="text-align:right;">
0.21836
</td>
<td style="text-align:right;">
0.22289
</td>
<td style="text-align:right;">
0.13018
</td>
<td style="text-align:right;">
0.22079
</td>
<td style="text-align:right;">
0.31075
</td>
<td style="text-align:right;">
0.16374
</td>
<td style="text-align:right;">
0.39075
</td>
<td style="text-align:right;">
0.21754
</td>
<td style="text-align:right;">
0.22446
</td>
<td style="text-align:right;">
0.08578
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_vulgatus_57955
</td>
<td style="text-align:right;">
2.75539
</td>
<td style="text-align:right;">
2.75647
</td>
<td style="text-align:right;">
2.75253
</td>
<td style="text-align:right;">
2.58013
</td>
<td style="text-align:right;">
1.04433
</td>
<td style="text-align:right;">
2.55823
</td>
<td style="text-align:right;">
0.31853
</td>
<td style="text-align:right;">
0.28593
</td>
<td style="text-align:right;">
0.36394
</td>
<td style="text-align:right;">
0.32930
</td>
<td style="text-align:right;">
0.28828
</td>
<td style="text-align:right;">
0.29281
</td>
<td style="text-align:right;">
0.32876
</td>
<td style="text-align:right;">
0.25751
</td>
<td style="text-align:right;">
0.28225
</td>
<td style="text-align:right;">
0.31390
</td>
<td style="text-align:right;">
0.08231
</td>
<td style="text-align:right;">
0.26564
</td>
<td style="text-align:right;">
0.29438
</td>
<td style="text-align:right;">
0.33126
</td>
</tr>
<tr>
<td style="text-align:left;">
Bacteroides_xylanisolvens_57185
</td>
<td style="text-align:right;">
2.80366
</td>
<td style="text-align:right;">
2.80474
</td>
<td style="text-align:right;">
2.80080
</td>
<td style="text-align:right;">
2.62840
</td>
<td style="text-align:right;">
1.09260
</td>
<td style="text-align:right;">
2.60650
</td>
<td style="text-align:right;">
0.05108
</td>
<td style="text-align:right;">
0.03594
</td>
<td style="text-align:right;">
0.18621
</td>
<td style="text-align:right;">
0.15157
</td>
<td style="text-align:right;">
0.05069
</td>
<td style="text-align:right;">
0.03294
</td>
<td style="text-align:right;">
0.15103
</td>
<td style="text-align:right;">
0.11298
</td>
<td style="text-align:right;">
0.20294
</td>
<td style="text-align:right;">
0.13617
</td>
<td style="text-align:right;">
0.28294
</td>
<td style="text-align:right;">
0.10973
</td>
<td style="text-align:right;">
0.01201
</td>
<td style="text-align:right;">
0.15353
</td>
</tr>
<tr>
<td style="text-align:left;">
Bifidobacterium_dentium_55719
</td>
<td style="text-align:right;">
1.20537
</td>
<td style="text-align:right;">
1.20645
</td>
<td style="text-align:right;">
1.20251
</td>
<td style="text-align:right;">
2.98735
</td>
<td style="text-align:right;">
3.15587
</td>
<td style="text-align:right;">
2.83737
</td>
<td style="text-align:right;">
3.10507
</td>
<td style="text-align:right;">
3.07247
</td>
<td style="text-align:right;">
3.15048
</td>
<td style="text-align:right;">
3.11584
</td>
<td style="text-align:right;">
3.07482
</td>
<td style="text-align:right;">
3.07935
</td>
<td style="text-align:right;">
3.11530
</td>
<td style="text-align:right;">
3.04405
</td>
<td style="text-align:right;">
3.06879
</td>
<td style="text-align:right;">
3.10044
</td>
<td style="text-align:right;">
3.02121
</td>
<td style="text-align:right;">
3.05218
</td>
<td style="text-align:right;">
3.08092
</td>
<td style="text-align:right;">
3.11780
</td>
</tr>
<tr>
<td style="text-align:left;">
Bifidobacterium_longum_57796
</td>
<td style="text-align:right;">
1.24494
</td>
<td style="text-align:right;">
1.24602
</td>
<td style="text-align:right;">
1.24208
</td>
<td style="text-align:right;">
3.02692
</td>
<td style="text-align:right;">
3.19544
</td>
<td style="text-align:right;">
2.87694
</td>
<td style="text-align:right;">
3.14464
</td>
<td style="text-align:right;">
3.11204
</td>
<td style="text-align:right;">
3.19005
</td>
<td style="text-align:right;">
3.15541
</td>
<td style="text-align:right;">
3.11439
</td>
<td style="text-align:right;">
3.11892
</td>
<td style="text-align:right;">
3.15487
</td>
<td style="text-align:right;">
3.08362
</td>
<td style="text-align:right;">
3.10836
</td>
<td style="text-align:right;">
3.14001
</td>
<td style="text-align:right;">
3.06078
</td>
<td style="text-align:right;">
3.09175
</td>
<td style="text-align:right;">
3.12049
</td>
<td style="text-align:right;">
3.15737
</td>
</tr>
<tr>
<td style="text-align:left;">
Bilophila_wadsworthia_57364
</td>
<td style="text-align:right;">
3.01421
</td>
<td style="text-align:right;">
3.01529
</td>
<td style="text-align:right;">
3.01135
</td>
<td style="text-align:right;">
2.83895
</td>
<td style="text-align:right;">
2.89951
</td>
<td style="text-align:right;">
2.81705
</td>
<td style="text-align:right;">
2.84871
</td>
<td style="text-align:right;">
2.81611
</td>
<td style="text-align:right;">
2.89412
</td>
<td style="text-align:right;">
2.85948
</td>
<td style="text-align:right;">
2.81846
</td>
<td style="text-align:right;">
2.82299
</td>
<td style="text-align:right;">
2.85894
</td>
<td style="text-align:right;">
2.78769
</td>
<td style="text-align:right;">
2.81243
</td>
<td style="text-align:right;">
2.84408
</td>
<td style="text-align:right;">
2.76485
</td>
<td style="text-align:right;">
2.79582
</td>
<td style="text-align:right;">
2.82456
</td>
<td style="text-align:right;">
2.86144
</td>
</tr>
<tr>
<td style="text-align:left;">
Blautia_wexlerae_56130
</td>
<td style="text-align:right;">
2.65501
</td>
<td style="text-align:right;">
2.65609
</td>
<td style="text-align:right;">
2.65215
</td>
<td style="text-align:right;">
2.60783
</td>
<td style="text-align:right;">
2.77635
</td>
<td style="text-align:right;">
0.72323
</td>
<td style="text-align:right;">
2.72555
</td>
<td style="text-align:right;">
2.69295
</td>
<td style="text-align:right;">
2.77096
</td>
<td style="text-align:right;">
2.73632
</td>
<td style="text-align:right;">
2.69530
</td>
<td style="text-align:right;">
2.69983
</td>
<td style="text-align:right;">
2.73578
</td>
<td style="text-align:right;">
2.66453
</td>
<td style="text-align:right;">
2.68927
</td>
<td style="text-align:right;">
2.72092
</td>
<td style="text-align:right;">
2.64169
</td>
<td style="text-align:right;">
2.67266
</td>
<td style="text-align:right;">
2.70140
</td>
<td style="text-align:right;">
2.73828
</td>
</tr>
<tr>
<td style="text-align:left;">
Campylobacter_concisus_62542
</td>
<td style="text-align:right;">
3.12324
</td>
<td style="text-align:right;">
3.12432
</td>
<td style="text-align:right;">
3.12038
</td>
<td style="text-align:right;">
2.94798
</td>
<td style="text-align:right;">
3.00854
</td>
<td style="text-align:right;">
2.92608
</td>
<td style="text-align:right;">
2.95774
</td>
<td style="text-align:right;">
2.92514
</td>
<td style="text-align:right;">
3.00315
</td>
<td style="text-align:right;">
2.96851
</td>
<td style="text-align:right;">
2.92749
</td>
<td style="text-align:right;">
2.93202
</td>
<td style="text-align:right;">
2.96797
</td>
<td style="text-align:right;">
2.89672
</td>
<td style="text-align:right;">
2.92146
</td>
<td style="text-align:right;">
2.95311
</td>
<td style="text-align:right;">
2.87388
</td>
<td style="text-align:right;">
2.90485
</td>
<td style="text-align:right;">
2.93359
</td>
<td style="text-align:right;">
2.97047
</td>
</tr>
<tr>
<td style="text-align:left;">
Clostridiaceae_bacterium_58376
</td>
<td style="text-align:right;">
2.66375
</td>
<td style="text-align:right;">
2.66483
</td>
<td style="text-align:right;">
2.66089
</td>
<td style="text-align:right;">
2.48849
</td>
<td style="text-align:right;">
2.54905
</td>
<td style="text-align:right;">
2.46659
</td>
<td style="text-align:right;">
2.49825
</td>
<td style="text-align:right;">
2.46565
</td>
<td style="text-align:right;">
2.54366
</td>
<td style="text-align:right;">
2.50902
</td>
<td style="text-align:right;">
2.46800
</td>
<td style="text-align:right;">
2.47253
</td>
<td style="text-align:right;">
2.50848
</td>
<td style="text-align:right;">
2.43723
</td>
<td style="text-align:right;">
2.46197
</td>
<td style="text-align:right;">
2.49362
</td>
<td style="text-align:right;">
2.41439
</td>
<td style="text-align:right;">
2.44536
</td>
<td style="text-align:right;">
2.47410
</td>
<td style="text-align:right;">
2.51098
</td>
</tr>
<tr>
<td style="text-align:left;">
Clostridiales_bacterium_55367
</td>
<td style="text-align:right;">
2.64215
</td>
<td style="text-align:right;">
2.64323
</td>
<td style="text-align:right;">
2.63929
</td>
<td style="text-align:right;">
2.59497
</td>
<td style="text-align:right;">
2.76349
</td>
<td style="text-align:right;">
0.71037
</td>
<td style="text-align:right;">
2.71269
</td>
<td style="text-align:right;">
2.68009
</td>
<td style="text-align:right;">
2.75810
</td>
<td style="text-align:right;">
2.72346
</td>
<td style="text-align:right;">
2.68244
</td>
<td style="text-align:right;">
2.68697
</td>
<td style="text-align:right;">
2.72292
</td>
<td style="text-align:right;">
2.65167
</td>
<td style="text-align:right;">
2.67641
</td>
<td style="text-align:right;">
2.70806
</td>
<td style="text-align:right;">
2.62883
</td>
<td style="text-align:right;">
2.65980
</td>
<td style="text-align:right;">
2.68854
</td>
<td style="text-align:right;">
2.72542
</td>
</tr>
<tr>
<td style="text-align:left;">
Clostridiales_bacterium_56470
</td>
<td style="text-align:right;">
2.74885
</td>
<td style="text-align:right;">
2.74993
</td>
<td style="text-align:right;">
2.74599
</td>
<td style="text-align:right;">
2.70167
</td>
<td style="text-align:right;">
2.87019
</td>
<td style="text-align:right;">
1.52633
</td>
<td style="text-align:right;">
2.81939
</td>
<td style="text-align:right;">
2.78679
</td>
<td style="text-align:right;">
2.86480
</td>
<td style="text-align:right;">
2.83016
</td>
<td style="text-align:right;">
2.78914
</td>
<td style="text-align:right;">
2.79367
</td>
<td style="text-align:right;">
2.82962
</td>
<td style="text-align:right;">
2.75837
</td>
<td style="text-align:right;">
2.78311
</td>
<td style="text-align:right;">
2.81476
</td>
<td style="text-align:right;">
2.73553
</td>
<td style="text-align:right;">
2.76650
</td>
<td style="text-align:right;">
2.79524
</td>
<td style="text-align:right;">
2.83212
</td>
</tr>
<tr>
<td style="text-align:left;">
Clostridiales_bacterium_58600
</td>
<td style="text-align:right;">
2.60480
</td>
<td style="text-align:right;">
2.60588
</td>
<td style="text-align:right;">
2.60194
</td>
<td style="text-align:right;">
2.55762
</td>
<td style="text-align:right;">
2.72614
</td>
<td style="text-align:right;">
1.53864
</td>
<td style="text-align:right;">
2.67534
</td>
<td style="text-align:right;">
2.64274
</td>
<td style="text-align:right;">
2.72075
</td>
<td style="text-align:right;">
2.68611
</td>
<td style="text-align:right;">
2.64509
</td>
<td style="text-align:right;">
2.64962
</td>
<td style="text-align:right;">
2.68557
</td>
<td style="text-align:right;">
2.61432
</td>
<td style="text-align:right;">
2.63906
</td>
<td style="text-align:right;">
2.67071
</td>
<td style="text-align:right;">
2.59148
</td>
<td style="text-align:right;">
2.62245
</td>
<td style="text-align:right;">
2.65119
</td>
<td style="text-align:right;">
2.68807
</td>
</tr>
<tr>
<td style="text-align:left;">
Clostridiales_bacterium_59661
</td>
<td style="text-align:right;">
2.56439
</td>
<td style="text-align:right;">
2.56547
</td>
<td style="text-align:right;">
2.56153
</td>
<td style="text-align:right;">
2.51721
</td>
<td style="text-align:right;">
2.68573
</td>
<td style="text-align:right;">
0.21769
</td>
<td style="text-align:right;">
2.63493
</td>
<td style="text-align:right;">
2.60233
</td>
<td style="text-align:right;">
2.68034
</td>
<td style="text-align:right;">
2.64570
</td>
<td style="text-align:right;">
2.60468
</td>
<td style="text-align:right;">
2.60921
</td>
<td style="text-align:right;">
2.64516
</td>
<td style="text-align:right;">
2.57391
</td>
<td style="text-align:right;">
2.59865
</td>
<td style="text-align:right;">
2.63030
</td>
<td style="text-align:right;">
2.55107
</td>
<td style="text-align:right;">
2.58204
</td>
<td style="text-align:right;">
2.61078
</td>
<td style="text-align:right;">
2.64766
</td>
</tr>
<tr>
<td style="text-align:left;">
Clostridiales_bacterium_59663
</td>
<td style="text-align:right;">
2.68495
</td>
<td style="text-align:right;">
2.68603
</td>
<td style="text-align:right;">
2.68209
</td>
<td style="text-align:right;">
2.63777
</td>
<td style="text-align:right;">
2.80629
</td>
<td style="text-align:right;">
0.75317
</td>
<td style="text-align:right;">
2.75549
</td>
<td style="text-align:right;">
2.72289
</td>
<td style="text-align:right;">
2.80090
</td>
<td style="text-align:right;">
2.76626
</td>
<td style="text-align:right;">
2.72524
</td>
<td style="text-align:right;">
2.72977
</td>
<td style="text-align:right;">
2.76572
</td>
<td style="text-align:right;">
2.69447
</td>
<td style="text-align:right;">
2.71921
</td>
<td style="text-align:right;">
2.75086
</td>
<td style="text-align:right;">
2.67163
</td>
<td style="text-align:right;">
2.70260
</td>
<td style="text-align:right;">
2.73134
</td>
<td style="text-align:right;">
2.76822
</td>
</tr>
<tr>
<td style="text-align:left;">
Clostridiales_bacterium_59664
</td>
<td style="text-align:right;">
2.71018
</td>
<td style="text-align:right;">
2.71126
</td>
<td style="text-align:right;">
2.70732
</td>
<td style="text-align:right;">
2.66300
</td>
<td style="text-align:right;">
2.83152
</td>
<td style="text-align:right;">
0.77840
</td>
<td style="text-align:right;">
2.78072
</td>
<td style="text-align:right;">
2.74812
</td>
<td style="text-align:right;">
2.82613
</td>
<td style="text-align:right;">
2.79149
</td>
<td style="text-align:right;">
2.75047
</td>
<td style="text-align:right;">
2.75500
</td>
<td style="text-align:right;">
2.79095
</td>
<td style="text-align:right;">
2.71970
</td>
<td style="text-align:right;">
2.74444
</td>
<td style="text-align:right;">
2.77609
</td>
<td style="text-align:right;">
2.69686
</td>
<td style="text-align:right;">
2.72783
</td>
<td style="text-align:right;">
2.75657
</td>
<td style="text-align:right;">
2.79345
</td>
</tr>
<tr>
<td style="text-align:left;">
Clostridiales_bacterium_61057
</td>
<td style="text-align:right;">
2.67920
</td>
<td style="text-align:right;">
2.68028
</td>
<td style="text-align:right;">
2.67634
</td>
<td style="text-align:right;">
2.63202
</td>
<td style="text-align:right;">
2.80054
</td>
<td style="text-align:right;">
0.74742
</td>
<td style="text-align:right;">
2.74974
</td>
<td style="text-align:right;">
2.71714
</td>
<td style="text-align:right;">
2.79515
</td>
<td style="text-align:right;">
2.76051
</td>
<td style="text-align:right;">
2.71949
</td>
<td style="text-align:right;">
2.72402
</td>
<td style="text-align:right;">
2.75997
</td>
<td style="text-align:right;">
2.68872
</td>
<td style="text-align:right;">
2.71346
</td>
<td style="text-align:right;">
2.74511
</td>
<td style="text-align:right;">
2.66588
</td>
<td style="text-align:right;">
2.69685
</td>
<td style="text-align:right;">
2.72559
</td>
<td style="text-align:right;">
2.76247
</td>
</tr>
<tr>
<td style="text-align:left;">
Clostridium_bartlettii_61535
</td>
<td style="text-align:right;">
2.41255
</td>
<td style="text-align:right;">
2.41363
</td>
<td style="text-align:right;">
2.40969
</td>
<td style="text-align:right;">
2.36537
</td>
<td style="text-align:right;">
2.53389
</td>
<td style="text-align:right;">
1.34639
</td>
<td style="text-align:right;">
2.48309
</td>
<td style="text-align:right;">
2.45049
</td>
<td style="text-align:right;">
2.52850
</td>
<td style="text-align:right;">
2.49386
</td>
<td style="text-align:right;">
2.45284
</td>
<td style="text-align:right;">
2.45737
</td>
<td style="text-align:right;">
2.49332
</td>
<td style="text-align:right;">
2.42207
</td>
<td style="text-align:right;">
2.44681
</td>
<td style="text-align:right;">
2.47846
</td>
<td style="text-align:right;">
2.39923
</td>
<td style="text-align:right;">
2.43020
</td>
<td style="text-align:right;">
2.45894
</td>
<td style="text-align:right;">
2.49582
</td>
</tr>
<tr>
<td style="text-align:left;">
Clostridium_bolteae_57158
</td>
<td style="text-align:right;">
2.63024
</td>
<td style="text-align:right;">
2.63132
</td>
<td style="text-align:right;">
2.62738
</td>
<td style="text-align:right;">
2.58306
</td>
<td style="text-align:right;">
2.75158
</td>
<td style="text-align:right;">
0.69846
</td>
<td style="text-align:right;">
2.70078
</td>
<td style="text-align:right;">
2.66818
</td>
<td style="text-align:right;">
2.74619
</td>
<td style="text-align:right;">
2.71155
</td>
<td style="text-align:right;">
2.67053
</td>
<td style="text-align:right;">
2.67506
</td>
<td style="text-align:right;">
2.71101
</td>
<td style="text-align:right;">
2.63976
</td>
<td style="text-align:right;">
2.66450
</td>
<td style="text-align:right;">
2.69615
</td>
<td style="text-align:right;">
2.61692
</td>
<td style="text-align:right;">
2.64789
</td>
<td style="text-align:right;">
2.67663
</td>
<td style="text-align:right;">
2.71351
</td>
</tr>
<tr>
<td style="text-align:left;">
Clostridium_clostridioforme_51842
</td>
<td style="text-align:right;">
2.63856
</td>
<td style="text-align:right;">
2.63964
</td>
<td style="text-align:right;">
2.63570
</td>
<td style="text-align:right;">
2.59138
</td>
<td style="text-align:right;">
2.75990
</td>
<td style="text-align:right;">
0.70678
</td>
<td style="text-align:right;">
2.70910
</td>
<td style="text-align:right;">
2.67650
</td>
<td style="text-align:right;">
2.75451
</td>
<td style="text-align:right;">
2.71987
</td>
<td style="text-align:right;">
2.67885
</td>
<td style="text-align:right;">
2.68338
</td>
<td style="text-align:right;">
2.71933
</td>
<td style="text-align:right;">
2.64808
</td>
<td style="text-align:right;">
2.67282
</td>
<td style="text-align:right;">
2.70447
</td>
<td style="text-align:right;">
2.62524
</td>
<td style="text-align:right;">
2.65621
</td>
<td style="text-align:right;">
2.68495
</td>
<td style="text-align:right;">
2.72183
</td>
</tr>
</tbody>
</table>
</div>
<p>Here is a heat map plot of the distance matrix:
<img src="microbiome_tutorial_files/figure-html/unnamed-chunk-37-1.png" width="768"></p>
<p>Here is a summary of all of the pairwise distances in the matrix:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">distance_matrix</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">#&gt;   0.000   2.298   2.612   2.397   2.858   3.736</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="convert-a-pairwise-phylogenetic-distance-matrix-to-a-pairwise-similarity-matrix">(2) Convert a pairwise phylogenetic distance matrix to a pairwise
similarity matrix<a class="anchor" aria-label="anchor" href="#convert-a-pairwise-phylogenetic-distance-matrix-to-a-pairwise-similarity-matrix"></a>
</h4>
<p>Next, we explore three transformations to convert the phylogenetic
distance between species
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>j</mi><annotation encoding="application/x-tex">j</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><annotation encoding="application/x-tex">D_{i,j}</annotation></semantics></math>,
to a similarity between species
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>j</mi><annotation encoding="application/x-tex">j</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>S</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><annotation encoding="application/x-tex">S_{i,j}</annotation></semantics></math>.
Each transformation must map distances of 0 to similarities of 1, and
very large distances to very small similarities. While there are many
possible transformations, we consider the following three:</p>
<ul>
<li><p>Difference:
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>S</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>=</mo><mn>1</mn><mo>−</mo><mfrac><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mrow><mo>max</mo><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac></mrow><annotation encoding="application/x-tex">S_{i,j}=1-\frac{D_{i,j}}{\max{(D_{i,j})}}</annotation></semantics></math></p></li>
<li><p>Exponential:
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>S</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>=</mo><msup><mi>e</mi><mrow><mo>−</mo><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub></mrow></msup></mrow><annotation encoding="application/x-tex">S_{i,j}=e^{-D_{i,j}}</annotation></semantics></math></p></li>
<li><p>Inverse:
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>S</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>=</mo><mfrac><mn>1</mn><mrow><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>+</mo><mn>1</mn></mrow></mfrac></mrow><annotation encoding="application/x-tex">S_{i,j}=\frac{1}{D_{i,j}+1}</annotation></semantics></math></p></li>
</ul>
<p>These three transformations are plotted below for the maximum
pairwise distance of our example data, approximately 3.7
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><mo>max</mo><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>≈</mo><mn>3.7</mn><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(\max{(D_{i,j})}\approx 3.7)</annotation></semantics></math>.
<img src="microbiome_tutorial_files/figure-html/unnamed-chunk-39-1.png" width="768"></p>
<p>We see from the above density plot and the summary at the end of the
preceding section that the phylogenetic distance between most species is
between about 2 and 3. In this region, the exponential transformation
yields much lower similarities than either the difference or the inverse
similarities.</p>
<p>Note that the relative shapes of these transformations changes as
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>max</mo><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\max{(D_{i,j})}</annotation></semantics></math>
increases. Consider the below plot, with
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>max</mo><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><mn>100</mn></mrow><annotation encoding="application/x-tex">\max{(D_{i,j})}=100</annotation></semantics></math>.
<img src="microbiome_tutorial_files/figure-html/unnamed-chunk-40-1.png" width="768"></p>
<p>The code for each transformation is quite simple:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">difference_similarity</span> <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">distance_matrix</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">distance_matrix</span><span class="op">)</span></span>
<span></span>
<span><span class="va">exponential_similarity</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span><span class="op">-</span><span class="va">distance_matrix</span><span class="op">)</span></span>
<span></span>
<span><span class="va">inverse_similarity</span> <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="op">(</span><span class="va">distance_matrix</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>These transformations result in very different pairwise similarity
matrices. Below, we visualize and summarize these matrices by:</p>
<p><strong>a.</strong> Plotting each of the three similarity matrices as
a heat map, with each pairwise similarity colored on a scale from 0
(dark purple) to 1 (bright yellow). Notice the consistent yellow
diagonal in each plot, confirming that each species has similarity 1
with itself, regardless of transformation.</p>
<p><strong>b.</strong> Plotting the distribution of pairwise similarity
values (all entries of each matrix) for each transformation.</p>
<p><strong>c.</strong> Computing summary statistics (minimum, first
quartile, median, mean, third quartile, maximum) across all pairwise
similarities for each transformation.</p>
<p><strong>(a) Heat maps</strong></p>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-42-1.png" width="768"><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-42-2.png" width="768"><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-42-3.png" width="768"></p>
<p><strong>(b) Distributions of pairwise similarities between species
for each transformation</strong>
<img src="microbiome_tutorial_files/figure-html/unnamed-chunk-43-1.png" width="768"></p>
<p><strong>(c) Summary statistics</strong></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">difference_similarity</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">#&gt;  0.0000  0.2352  0.3010  0.3583  0.3850  1.0000</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">exponential_similarity</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">#&gt; 0.02384 0.05740 0.07341 0.13890 0.10048 1.00000</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">inverse_similarity</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">#&gt;  0.2111  0.2592  0.2769  0.3247  0.3032  1.0000</span></span></code></pre></div>
<p>These summaries illuminate important differences among the
transformations.</p>
<ul>
<li>
<p>The <strong>difference</strong> transformation generally results
in <em>high</em> pairwise similarities.</p>
<ul>
<li><p>The heat map (a) is dominated by turquoise (values between 0.25
and 0.5), with bands of purple (groups of species with low similarity to
most other species) and blocks of yellow (groups of species with high
similarity to one another).</p></li>
<li><p>The distribution of pairwise similarities (b) has a broad peak
between 0.2 and 0.4, and a heavy tail (i.e., high similarity values are
fairly abundant).</p></li>
<li><p>The mean similarity between species on a scale of 0 to 1 (c) is
<strong>0.36.</strong></p></li>
</ul>
</li>
<li>
<p>The <strong>exponential</strong> transformation generally results
in <em>low</em> pairwise similarities.</p>
<ul>
<li><p>The heat map (a) is dominated by dark purple (similarity values
close to 0), except for small blocks of high similarity, corresponding
to closely related species.</p></li>
<li><p>The distribution of pairwise similarities (b) has a narrow peak
between 0 and 0.1.</p></li>
<li><p>The mean similarity between species on a scale of 0 to 1 (c) is
just <strong>0.14.</strong></p></li>
</ul>
</li>
<li>
<p>The <strong>inverse</strong> transformation is similar to the
difference distribution for the range of phylogenetic distances present
in this data set
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>∈</mo><mrow><mo stretchy="true" form="prefix">[</mo><mn>0</mn><mo>,</mo><mn>3.7</mn><mo stretchy="true" form="postfix">]</mo></mrow></mrow><annotation encoding="application/x-tex">D_{i,j}\in [0,3.7]</annotation></semantics></math>),
but the transformation will be more similar to the exponential
distribution when the maximum phylogenetic distance
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>max</mo><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\max{(D_{i,j})}</annotation></semantics></math>)
is larger (e.g., see the above plot of the transformation functions when
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>max</mo><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><mn>100</mn></mrow><annotation encoding="application/x-tex">\max{(D_{i,j})}=100</annotation></semantics></math>).</p>
<ul>
<li><p>Its distribution of pairwise similarities (b) has a narrow peak
like the exponential transformation, but a center around 0.3 like the
difference transformation.</p></li>
<li><p>This shift is also reflected by its large mean similarity value
of <strong>0.32</strong>, which is similar to that of the difference
transformation (c).</p></li>
</ul>
</li>
</ul>
<p>Your choice of transformation depends on your desired distribution of
pairwise similarities:</p>
<ul>
<li><p>Use the <strong>difference</strong> transformation if you want to
account for the relatedness of distantly related species in addition to
closely related species. This transformation is also the only of these
transformations that is linear, so the shape of the distribution of
similarities (b) is just a mirror image of the distribution of pairwise
distances (the first plot of this sub-section). It is also the only
transformation whose distribution of similarities does not change with
the maximum phylogenetic distance
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>max</mo><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\max{(D_{i,j})}</annotation></semantics></math>).</p></li>
<li><p>Use the <strong>exponential</strong> transformation if you want
only very closely related species to be treated as similar.</p></li>
<li><p>Use the <strong>inverse</strong> transformation if you want an
intermediate between the difference and exponential transformations,
though note that the shape depends on the value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>max</mo><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>D</mi><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\max{(D_{i,j})}</annotation></semantics></math>.</p></li>
</ul>
<p>For the example analysis in this tutorial and in the accompanying
paper, we use the exponential transformation because we wish to only
account for similarities among very closely related species. However,
this decision is informed by the biological processes at play in the
data, so other choices may be more suitable for different data sets.</p>
</div>
<div class="section level4">
<h4 id="how-do-these-transformations-influence-the-value-of-fava">(3) How do these transformations influence the value of FAVA?<a class="anchor" aria-label="anchor" href="#how-do-these-transformations-influence-the-value-of-fava"></a>
</h4>
<p>We first compute FAVA across all samples for each subject using each
of the three different similarity matrices (difference, exponential, and
inverse). In each of these calculations, we weight samples based on
their sampling time.</p>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-45-1.png" width="768"></p>
<p>We find that the exponential and inverse transformations yield very
similar results. The difference transformation, on the other hand,
yields FAVA values much lower than the other transformations for subject
XDA and XMA, though not for XBA.</p>
<p>We next repeat the sliding window analysis for each of the three
transformations.</p>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-46-1.png" width="768"></p>
<p>We find that the results across transformations are qualitatively the
same, despite small differences. Regardless of the transformation used
to generate the similarity matrix, XBA has the highest peak variability
and does not stabilize, while XDA and XMA both stabilize at FAVA values
similar to their initial values.</p>
<p>In order to more easily compare the three transformations, we plot
each sliding window as a point, with the location along the x-axis
corresponding to the center of the sliding window.</p>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-47-1.png" width="768"></p>
<p>We find that the relative ordering of FAVA values for the three
transformations depends on both subject and the window time relative to
antibiotic. For windows overlapping with the antibiotic period (grey
rectangle), the FAVA values generated from a difference-transformed
similarity matrix are very different from those generated by the other
two transformations—greater for subject XBA, and smaller for subjects
XDA and XMA. For windows far from the antibiotic window, the results for
each transformation are more similar.</p>
<!-- **Maike to remove below section after talking with Katherine** -->
<!-- I speculated that there could be an association between the amount of variability in a window and the disparity between methods (perhaps when there's more variability, the methods disagree more?), but the results are extremely inconsistent among subjects. -->
<!-- ```{r} -->
<!-- fava_window_transformation %>% pivot_wider(names_from = Transformation, values_from = FAVA) %>% -->
<!--   mutate("Difference - Exponential" = Difference - Exponential) %>% -->
<!--   ggplot(aes(x = Exponential, y = `Difference - Exponential`, color = abs((w6 + w1)/2 - 31.5))) + -->
<!--   geom_smooth(method = "lm", color = "grey", alpha = 0.3) + -->
<!--   geom_point(size = 3, alpha = 0.8) + theme_bw() + -->
<!--   facet_wrap(~ group, scales = "free") + -->
<!--   scale_color_viridis_c(name = "Days from\nabx center") -->
<!-- ``` -->
</div>
</div>
<div class="section level3">
<h3 id="when-to-normalize-fava">When to normalize FAVA<a class="anchor" aria-label="anchor" href="#when-to-normalize-fava"></a>
</h3>
<p>Recall that FAVA uses the population-genetic statistic
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
to quantify variability across many samples of microbial community
composition.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><mo>=</mo><mn>0</mn></mrow><annotation encoding="application/x-tex">F_{ST}=0</annotation></semantics></math>
when every sample is identical and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><mo>=</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">F_{ST}=1</annotation></semantics></math>
when each sample is comprised entirely of a single taxon and there are
at least two distinct taxa present across all of the samples.</p>
<p>In this section, we explore how
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
is constrained when the sample size is small, and how we can account for
this constraint by normalizing
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
by its theoretical upper bound conditional on the sample size,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>I</mi><annotation encoding="application/x-tex">I</annotation></semantics></math>,
and the mean abundance of the most abundant taxon,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>.</p>
<p><strong>In short, if you wish to compare the variability of relative
abundance matrices with few rows/samples (I) and very different mean
abundances of the dominant taxon (M), you may wish to normalize
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
by its theoretical upper bound in order to avoid variability differences
driven by the difference in M.</strong></p>
<div class="section level4">
<h4 id="when-i-is-small-which-relative-abundance-matrices-have-f_st1">When
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>I</mi><annotation encoding="application/x-tex">I</annotation></semantics></math>
is small, which relative abundance matrices have
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><mo>=</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">F_{ST}=1</annotation></semantics></math>?<a class="anchor" aria-label="anchor" href="#when-i-is-small-which-relative-abundance-matrices-have-f_st1"></a>
</h4>
<p>For a relative abundance matrix with just two samples
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>I</mi><mo>=</mo><mn>2</mn></mrow><annotation encoding="application/x-tex">I=2</annotation></semantics></math>),
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
can reach its upper bound of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>1</mn><annotation encoding="application/x-tex">1</annotation></semantics></math>
only when each sample is entirely comprised of a different species:</p>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-48-1.png" width="480"></p>
<p>In this case, the mean abundance of the most abundant species across
both samples is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>M</mi><mo>=</mo><mfrac><mn>1</mn><mn>2</mn></mfrac></mrow><annotation encoding="application/x-tex">M=\frac{1}{2}</annotation></semantics></math>.
This suggests that, if there are two samples
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>I</mi><mo>=</mo><mn>2</mn></mrow><annotation encoding="application/x-tex">I=2</annotation></semantics></math>),
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
can only equal
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>1</mn><annotation encoding="application/x-tex">1</annotation></semantics></math>
if
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>M</mi><mo>=</mo><mfrac><mn>1</mn><mn>2</mn></mfrac></mrow><annotation encoding="application/x-tex">M=\frac{1}{2}</annotation></semantics></math>.</p>
<p>For a relative abundance matrix with three samples
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>I</mi><mo>=</mo><mn>3</mn></mrow><annotation encoding="application/x-tex">I=3</annotation></semantics></math>),
there are two relative abundance matrices that reach
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><mo>=</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">F_{ST}=1</annotation></semantics></math>:</p>
<ol style="list-style-type: decimal">
<li><p>Matrix A, in which each sample is comprised of a distinct
taxon</p></li>
<li><p>Matrix B, in which two samples are comprised of the same taxon
and one is comprised of a different taxon</p></li>
</ol>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-49-1.png" width="672"></p>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-50-1.png" width="576"></p>
<p>Here, the mean abundance of the most abundant taxon across all three
samples is either
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>M</mi><mo>=</mo><mfrac><mn>1</mn><mn>3</mn></mfrac></mrow><annotation encoding="application/x-tex">M=\frac{1}{3}</annotation></semantics></math>
(Matrix A) or
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>M</mi><mo>=</mo><mfrac><mn>2</mn><mn>3</mn></mfrac></mrow><annotation encoding="application/x-tex">M=\frac{2}{3}</annotation></semantics></math>
(Matrix B). It follows that, when
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>I</mi><mo>=</mo><mn>3</mn></mrow><annotation encoding="application/x-tex">I=3</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
can only equal
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>1</mn><annotation encoding="application/x-tex">1</annotation></semantics></math>
when
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>M</mi><mo>=</mo><mfrac><mn>1</mn><mn>3</mn></mfrac></mrow><annotation encoding="application/x-tex">M=\frac{1}{3}</annotation></semantics></math>
or
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mfrac><mn>2</mn><mn>3</mn></mfrac><annotation encoding="application/x-tex">\frac{2}{3}</annotation></semantics></math>.
For other values of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>,
the maximum possible value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
is less than
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>1</mn><annotation encoding="application/x-tex">1</annotation></semantics></math>.</p>
</div>
<div class="section level4">
<h4 id="in-general-what-is-the-upper-bound-on-f_st-conditional-on-m-and-i">In general, what is the upper bound on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
conditional on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>I</mi><annotation encoding="application/x-tex">I</annotation></semantics></math>?<a class="anchor" aria-label="anchor" href="#in-general-what-is-the-upper-bound-on-f_st-conditional-on-m-and-i"></a>
</h4>
<p>Previous work has derived a general equation for the upper bound on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
as a function of the number of samples
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>I</mi><annotation encoding="application/x-tex">I</annotation></semantics></math>)
and the abundance of the most abundant taxon
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>)
(<a href="https://rosenberglab.stanford.edu/papers/AlcalaRosenberg2022-PhilTransB.pdf" class="external-link">Alcala
&amp; Rosenberg 2022</a>). We can divide
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
by this upper bound in order to generate a variability statistic,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><mi>/</mi><msubsup><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow><mrow><mi>m</mi><mi>a</mi><mi>x</mi></mrow></msubsup></mrow><annotation encoding="application/x-tex">F_{ST}/F_{ST}^{max}</annotation></semantics></math>,
that ranges between 0 and 1 no matter what the value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>
is. In a previous project, we used
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><mi>/</mi><msubsup><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow><mrow><mi>m</mi><mi>a</mi><mi>x</mi></mrow></msubsup></mrow><annotation encoding="application/x-tex">F_{ST}/F_{ST}^{max}</annotation></semantics></math>
instead of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
because we didn’t want our results to be confounded by differences in
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>
(<a href="https://rosenberglab.stanford.edu/papers/MorrisonEtAl2022-MolEcolResources.pdf" class="external-link">Morrison,
Alcala, &amp; Rosenberg 2022</a>).</p>
<p>We plot the upper bound on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
as a function of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>
for different values of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>I</mi><annotation encoding="application/x-tex">I</annotation></semantics></math>
below.</p>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-51-1.png" width="1056"></p>
<p>Note that, as we expect from the preceding examples, for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>I</mi><mo>=</mo><mn>2</mn></mrow><annotation encoding="application/x-tex">I=2</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow><mrow><mi>m</mi><mi>a</mi><mi>x</mi></mrow></msubsup><mo>=</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">F_{ST}^{max}=1</annotation></semantics></math>
only when
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>M</mi><mo>=</mo><mfrac><mn>1</mn><mn>2</mn></mfrac></mrow><annotation encoding="application/x-tex">M=\frac{1}{2}</annotation></semantics></math>
and, for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>I</mi><mo>=</mo><mn>3</mn></mrow><annotation encoding="application/x-tex">I=3</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow><mrow><mi>m</mi><mi>a</mi><mi>x</mi></mrow></msubsup><mo>=</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">F_{ST}^{max}=1</annotation></semantics></math>
only when
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>M</mi><mo>=</mo><mfrac><mn>1</mn><mn>3</mn></mfrac></mrow><annotation encoding="application/x-tex">M=\frac{1}{3}</annotation></semantics></math>
or
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mfrac><mn>2</mn><mn>3</mn></mfrac><annotation encoding="application/x-tex">\frac{2}{3}</annotation></semantics></math>.</p>
<p>In general,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
is more constrained by
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>
when the sample size,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>I</mi><annotation encoding="application/x-tex">I</annotation></semantics></math>,
is small. In regions where the upper bound on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
is much less than
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>1</mn><annotation encoding="application/x-tex">1</annotation></semantics></math>,
the value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>
has a big influence on the value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>.
It can therefore be difficult to compare two relative abundance matrices
that have small sample sizes and very different values of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>.
In this situation, a difference in their values of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
could be driven mainly by their very different bounds on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
as a function of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>.
For this reason, you may wish to normalize
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
by this upper bound when making such a comparison.</p>
</div>
<div class="section level4">
<h4 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h4>
<p>Consider the following two matrices:</p>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-52-1.png" width="672"><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-52-2.png" width="672"></p>
<p>The
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
of matrix C is almost twice that of matrix D:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/fava.html">fava</a></span><span class="op">(</span><span class="va">C</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.1776815</span></span>
<span><span class="fu"><a href="../reference/fava.html">fava</a></span><span class="op">(</span><span class="va">D</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.09185804</span></span></code></pre></div>
<p>However, matrix C has a frequency of the most abundant taxon close to
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mfrac><mn>2</mn><mn>3</mn></mfrac><annotation encoding="application/x-tex">\frac{2}{3}</annotation></semantics></math>,
while the frequency of D’s most abundant taxon is much higher:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colMeans</a></span><span class="op">(</span><span class="va">C</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.65</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colMeans</a></span><span class="op">(</span><span class="va">D</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.85</span></span></code></pre></div>
<p>Because matrices C and D each have just
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>I</mi><mo>=</mo><mn>3</mn></mrow><annotation encoding="application/x-tex">I=3</annotation></semantics></math>
samples, the mean abundance of the most abundant species,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>,
influences the maximum value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
that each could attain. Matrix C has a value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>
that allows for a value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
close to 1, while matrix D is constrained to smaller
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
values:</p>
<p><img src="microbiome_tutorial_files/figure-html/unnamed-chunk-55-1.png" width="576"></p>
<p>The length of each colored vertical bar represents the value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msubsup><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow><mrow><mi>m</mi><mi>a</mi><mi>x</mi></mrow></msubsup><annotation encoding="application/x-tex">F_{ST}^{max}</annotation></semantics></math>
for each matrix. Since it is much closer to 1 for C than for D, the
value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
for matrix C will be less influenced by the normalization. Indeed, when
we divide each matrix’s value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><annotation encoding="application/x-tex">F_{ST}</annotation></semantics></math>
by its upper bound conditional on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>I</mi><annotation encoding="application/x-tex">I</annotation></semantics></math>,
we find that D has a larger value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow></msub><mi>/</mi><msubsup><mi>F</mi><mrow><mi>S</mi><mi>T</mi></mrow><mrow><mi>m</mi><mi>a</mi><mi>x</mi></mrow></msubsup></mrow><annotation encoding="application/x-tex">F_{ST}/F_{ST}^{max}</annotation></semantics></math>
than matrix C.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/fava.html">fava</a></span><span class="op">(</span><span class="va">C</span>, normalized <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.1906327</span></span>
<span><span class="fu"><a href="../reference/fava.html">fava</a></span><span class="op">(</span><span class="va">D</span>, normalized <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.2602644</span></span></code></pre></div>
<p>This change suggests that the difference in variability between
matrices C and D is heavily influenced by the difference in the
abundance of their dominant taxa.</p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Maike Morrison.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
